// @vercel/sdk/esm/models/additionalcontactinforequired.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const AdditionalContactInfoRequiredCode = {
    AdditionalContactInfoRequired: "additional_contact_info_required",
};
/**
 * Additional contact information is required for the TLD.
 */
export class AdditionalContactInfoRequired extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "AdditionalContactInfoRequired";
    }
}
/** @internal */
export const AdditionalContactInfoRequiredCode$inboundSchema = z.nativeEnum(AdditionalContactInfoRequiredCode);
/** @internal */
export const AdditionalContactInfoRequiredCode$outboundSchema = AdditionalContactInfoRequiredCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var AdditionalContactInfoRequiredCode$;
(function (AdditionalContactInfoRequiredCode$) {
    /** @deprecated use `AdditionalContactInfoRequiredCode$inboundSchema` instead. */
    AdditionalContactInfoRequiredCode$.inboundSchema = AdditionalContactInfoRequiredCode$inboundSchema;
    /** @deprecated use `AdditionalContactInfoRequiredCode$outboundSchema` instead. */
    AdditionalContactInfoRequiredCode$.outboundSchema = AdditionalContactInfoRequiredCode$outboundSchema;
})(AdditionalContactInfoRequiredCode$ || (AdditionalContactInfoRequiredCode$ = {}));
/** @internal */
export const AdditionalContactInfoRequired$inboundSchema = z.object({
    status: z.number(),
    code: AdditionalContactInfoRequiredCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new AdditionalContactInfoRequired(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const AdditionalContactInfoRequired$outboundSchema = z.instanceof(AdditionalContactInfoRequired)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: AdditionalContactInfoRequiredCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var AdditionalContactInfoRequired$;
(function (AdditionalContactInfoRequired$) {
    /** @deprecated use `AdditionalContactInfoRequired$inboundSchema` instead. */
    AdditionalContactInfoRequired$.inboundSchema = AdditionalContactInfoRequired$inboundSchema;
    /** @deprecated use `AdditionalContactInfoRequired$outboundSchema` instead. */
    AdditionalContactInfoRequired$.outboundSchema = AdditionalContactInfoRequired$outboundSchema;
})(AdditionalContactInfoRequired$ || (AdditionalContactInfoRequired$ = {}));
