// @vercel/sdk/esm/models/badrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const BadRequestCode = {
    BadRequest: "bad_request",
};
export class BadRequest extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "BadRequest";
    }
}
/** @internal */
export const BadRequestCode$inboundSchema = z.nativeEnum(BadRequestCode);
/** @internal */
export const BadRequestCode$outboundSchema = BadRequestCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BadRequestCode$;
(function (BadRequestCode$) {
    /** @deprecated use `BadRequestCode$inboundSchema` instead. */
    BadRequestCode$.inboundSchema = BadRequestCode$inboundSchema;
    /** @deprecated use `BadRequestCode$outboundSchema` instead. */
    BadRequestCode$.outboundSchema = BadRequestCode$outboundSchema;
})(BadRequestCode$ || (BadRequestCode$ = {}));
/** @internal */
export const BadRequest$inboundSchema = z.object({
    status: z.number(),
    code: BadRequestCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new BadRequest(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const BadRequest$outboundSchema = z.instanceof(BadRequest)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: BadRequestCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BadRequest$;
(function (BadRequest$) {
    /** @deprecated use `BadRequest$inboundSchema` instead. */
    BadRequest$.inboundSchema = BadRequest$inboundSchema;
    /** @deprecated use `BadRequest$outboundSchema` instead. */
    BadRequest$.outboundSchema = BadRequest$outboundSchema;
})(BadRequest$ || (BadRequest$ = {}));
