// @vercel/sdk/esm/models/buysingledomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AdditionalContactInfoRequired$inboundSchema, AdditionalContactInfoRequired$outboundSchema, } from "./additionalcontactinforequired.js";
import { DomainNotAvailable$inboundSchema, DomainNotAvailable$outboundSchema, } from "./domainnotavailable.js";
import { ExpectedPriceMismatch$inboundSchema, ExpectedPriceMismatch$outboundSchema, } from "./expectedpricemismatch.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { InvalidAdditionalContactInfo$inboundSchema, InvalidAdditionalContactInfo$outboundSchema, } from "./invalidadditionalcontactinfo.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
import { OrderTooExpensive$inboundSchema, OrderTooExpensive$outboundSchema, } from "./ordertooexpensive.js";
import { TldNotSupported$inboundSchema, TldNotSupported$outboundSchema, } from "./tldnotsupported.js";
export const BuySingleDomainMethod = {
    Get: "GET",
    Post: "POST",
    Put: "PUT",
    Delete: "DELETE",
    Patch: "PATCH",
};
/** @internal */
export const Additional$inboundSchema = z.object({});
/** @internal */
export const Additional$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Additional$;
(function (Additional$) {
    /** @deprecated use `Additional$inboundSchema` instead. */
    Additional$.inboundSchema = Additional$inboundSchema;
    /** @deprecated use `Additional$outboundSchema` instead. */
    Additional$.outboundSchema = Additional$outboundSchema;
})(Additional$ || (Additional$ = {}));
export function additionalToJSON(additional) {
    return JSON.stringify(Additional$outboundSchema.parse(additional));
}
export function additionalFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Additional$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Additional' from JSON`);
}
/** @internal */
export const ContactInformation$inboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    email: z.string(),
    phone: z.string(),
    address1: z.string(),
    address2: z.string().optional(),
    city: z.string(),
    state: z.string(),
    zip: z.string(),
    country: z.string(),
    companyName: z.string().optional(),
    fax: z.string().optional(),
    additional: z.lazy(() => Additional$inboundSchema).optional(),
});
/** @internal */
export const ContactInformation$outboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    email: z.string(),
    phone: z.string(),
    address1: z.string(),
    address2: z.string().optional(),
    city: z.string(),
    state: z.string(),
    zip: z.string(),
    country: z.string(),
    companyName: z.string().optional(),
    fax: z.string().optional(),
    additional: z.lazy(() => Additional$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ContactInformation$;
(function (ContactInformation$) {
    /** @deprecated use `ContactInformation$inboundSchema` instead. */
    ContactInformation$.inboundSchema = ContactInformation$inboundSchema;
    /** @deprecated use `ContactInformation$outboundSchema` instead. */
    ContactInformation$.outboundSchema = ContactInformation$outboundSchema;
})(ContactInformation$ || (ContactInformation$ = {}));
export function contactInformationToJSON(contactInformation) {
    return JSON.stringify(ContactInformation$outboundSchema.parse(contactInformation));
}
export function contactInformationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ContactInformation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ContactInformation' from JSON`);
}
/** @internal */
export const BuySingleDomainRequestBody$inboundSchema = z.object({
    autoRenew: z.boolean(),
    years: z.number(),
    expectedPrice: z.number(),
    contactInformation: z.lazy(() => ContactInformation$inboundSchema),
});
/** @internal */
export const BuySingleDomainRequestBody$outboundSchema = z.object({
    autoRenew: z.boolean(),
    years: z.number(),
    expectedPrice: z.number(),
    contactInformation: z.lazy(() => ContactInformation$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BuySingleDomainRequestBody$;
(function (BuySingleDomainRequestBody$) {
    /** @deprecated use `BuySingleDomainRequestBody$inboundSchema` instead. */
    BuySingleDomainRequestBody$.inboundSchema = BuySingleDomainRequestBody$inboundSchema;
    /** @deprecated use `BuySingleDomainRequestBody$outboundSchema` instead. */
    BuySingleDomainRequestBody$.outboundSchema = BuySingleDomainRequestBody$outboundSchema;
})(BuySingleDomainRequestBody$ || (BuySingleDomainRequestBody$ = {}));
export function buySingleDomainRequestBodyToJSON(buySingleDomainRequestBody) {
    return JSON.stringify(BuySingleDomainRequestBody$outboundSchema.parse(buySingleDomainRequestBody));
}
export function buySingleDomainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BuySingleDomainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BuySingleDomainRequestBody' from JSON`);
}
/** @internal */
export const BuySingleDomainRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    RequestBody: z.lazy(() => BuySingleDomainRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const BuySingleDomainRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    requestBody: z.lazy(() => BuySingleDomainRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BuySingleDomainRequest$;
(function (BuySingleDomainRequest$) {
    /** @deprecated use `BuySingleDomainRequest$inboundSchema` instead. */
    BuySingleDomainRequest$.inboundSchema = BuySingleDomainRequest$inboundSchema;
    /** @deprecated use `BuySingleDomainRequest$outboundSchema` instead. */
    BuySingleDomainRequest$.outboundSchema = BuySingleDomainRequest$outboundSchema;
})(BuySingleDomainRequest$ || (BuySingleDomainRequest$ = {}));
export function buySingleDomainRequestToJSON(buySingleDomainRequest) {
    return JSON.stringify(BuySingleDomainRequest$outboundSchema.parse(buySingleDomainRequest));
}
export function buySingleDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BuySingleDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BuySingleDomainRequest' from JSON`);
}
/** @internal */
export const BuySingleDomainDomainsRegistrarResponseResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const BuySingleDomainDomainsRegistrarResponseResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BuySingleDomainDomainsRegistrarResponseResponseBody$;
(function (BuySingleDomainDomainsRegistrarResponseResponseBody$) {
    /** @deprecated use `BuySingleDomainDomainsRegistrarResponseResponseBody$inboundSchema` instead. */
    BuySingleDomainDomainsRegistrarResponseResponseBody$.inboundSchema = BuySingleDomainDomainsRegistrarResponseResponseBody$inboundSchema;
    /** @deprecated use `BuySingleDomainDomainsRegistrarResponseResponseBody$outboundSchema` instead. */
    BuySingleDomainDomainsRegistrarResponseResponseBody$.outboundSchema = BuySingleDomainDomainsRegistrarResponseResponseBody$outboundSchema;
})(BuySingleDomainDomainsRegistrarResponseResponseBody$ || (BuySingleDomainDomainsRegistrarResponseResponseBody$ = {}));
export function buySingleDomainDomainsRegistrarResponseResponseBodyToJSON(buySingleDomainDomainsRegistrarResponseResponseBody) {
    return JSON.stringify(BuySingleDomainDomainsRegistrarResponseResponseBody$outboundSchema.parse(buySingleDomainDomainsRegistrarResponseResponseBody));
}
export function buySingleDomainDomainsRegistrarResponseResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BuySingleDomainDomainsRegistrarResponseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BuySingleDomainDomainsRegistrarResponseResponseBody' from JSON`);
}
/** @internal */
export const BuySingleDomainDomainsRegistrarResponseBody$inboundSchema = z.union([
    OrderTooExpensive$inboundSchema,
    InvalidAdditionalContactInfo$inboundSchema,
    AdditionalContactInfoRequired$inboundSchema,
    ExpectedPriceMismatch$inboundSchema,
    DomainNotAvailable$inboundSchema,
    TldNotSupported$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const BuySingleDomainDomainsRegistrarResponseBody$outboundSchema = z.union([
    OrderTooExpensive$outboundSchema,
    InvalidAdditionalContactInfo$outboundSchema,
    AdditionalContactInfoRequired$outboundSchema,
    ExpectedPriceMismatch$outboundSchema,
    DomainNotAvailable$outboundSchema,
    TldNotSupported$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BuySingleDomainDomainsRegistrarResponseBody$;
(function (BuySingleDomainDomainsRegistrarResponseBody$) {
    /** @deprecated use `BuySingleDomainDomainsRegistrarResponseBody$inboundSchema` instead. */
    BuySingleDomainDomainsRegistrarResponseBody$.inboundSchema = BuySingleDomainDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `BuySingleDomainDomainsRegistrarResponseBody$outboundSchema` instead. */
    BuySingleDomainDomainsRegistrarResponseBody$.outboundSchema = BuySingleDomainDomainsRegistrarResponseBody$outboundSchema;
})(BuySingleDomainDomainsRegistrarResponseBody$ || (BuySingleDomainDomainsRegistrarResponseBody$ = {}));
export function buySingleDomainDomainsRegistrarResponseBodyToJSON(buySingleDomainDomainsRegistrarResponseBody) {
    return JSON.stringify(BuySingleDomainDomainsRegistrarResponseBody$outboundSchema.parse(buySingleDomainDomainsRegistrarResponseBody));
}
export function buySingleDomainDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BuySingleDomainDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BuySingleDomainDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const BuySingleDomainMethod$inboundSchema = z.nativeEnum(BuySingleDomainMethod);
/** @internal */
export const BuySingleDomainMethod$outboundSchema = BuySingleDomainMethod$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BuySingleDomainMethod$;
(function (BuySingleDomainMethod$) {
    /** @deprecated use `BuySingleDomainMethod$inboundSchema` instead. */
    BuySingleDomainMethod$.inboundSchema = BuySingleDomainMethod$inboundSchema;
    /** @deprecated use `BuySingleDomainMethod$outboundSchema` instead. */
    BuySingleDomainMethod$.outboundSchema = BuySingleDomainMethod$outboundSchema;
})(BuySingleDomainMethod$ || (BuySingleDomainMethod$ = {}));
/** @internal */
export const Links$inboundSchema = z
    .object({
    href: z.string(),
    method: BuySingleDomainMethod$inboundSchema,
});
/** @internal */
export const Links$outboundSchema = z.object({
    href: z.string(),
    method: BuySingleDomainMethod$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Links$;
(function (Links$) {
    /** @deprecated use `Links$inboundSchema` instead. */
    Links$.inboundSchema = Links$inboundSchema;
    /** @deprecated use `Links$outboundSchema` instead. */
    Links$.outboundSchema = Links$outboundSchema;
})(Links$ || (Links$ = {}));
export function linksToJSON(links) {
    return JSON.stringify(Links$outboundSchema.parse(links));
}
export function linksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Links$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Links' from JSON`);
}
/** @internal */
export const BuySingleDomainResponseBody$inboundSchema = z.object({
    orderId: z.string(),
    _links: z.record(z.lazy(() => Links$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "_links": "links",
    });
});
/** @internal */
export const BuySingleDomainResponseBody$outboundSchema = z.object({
    orderId: z.string(),
    links: z.record(z.lazy(() => Links$outboundSchema)),
}).transform((v) => {
    return remap$(v, {
        links: "_links",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BuySingleDomainResponseBody$;
(function (BuySingleDomainResponseBody$) {
    /** @deprecated use `BuySingleDomainResponseBody$inboundSchema` instead. */
    BuySingleDomainResponseBody$.inboundSchema = BuySingleDomainResponseBody$inboundSchema;
    /** @deprecated use `BuySingleDomainResponseBody$outboundSchema` instead. */
    BuySingleDomainResponseBody$.outboundSchema = BuySingleDomainResponseBody$outboundSchema;
})(BuySingleDomainResponseBody$ || (BuySingleDomainResponseBody$ = {}));
export function buySingleDomainResponseBodyToJSON(buySingleDomainResponseBody) {
    return JSON.stringify(BuySingleDomainResponseBody$outboundSchema.parse(buySingleDomainResponseBody));
}
export function buySingleDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BuySingleDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BuySingleDomainResponseBody' from JSON`);
}
