// @vercel/sdk/esm/models/dangerouslydeletebytagsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const DangerouslyDeleteByTagsTarget = {
    Production: "production",
    Preview: "preview",
};
/** @internal */
export const DangerouslyDeleteByTagsTags$inboundSchema = z.union([z.array(z.string()), z.string()]);
/** @internal */
export const DangerouslyDeleteByTagsTags$outboundSchema = z.union([z.array(z.string()), z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DangerouslyDeleteByTagsTags$;
(function (DangerouslyDeleteByTagsTags$) {
    /** @deprecated use `DangerouslyDeleteByTagsTags$inboundSchema` instead. */
    DangerouslyDeleteByTagsTags$.inboundSchema = DangerouslyDeleteByTagsTags$inboundSchema;
    /** @deprecated use `DangerouslyDeleteByTagsTags$outboundSchema` instead. */
    DangerouslyDeleteByTagsTags$.outboundSchema = DangerouslyDeleteByTagsTags$outboundSchema;
})(DangerouslyDeleteByTagsTags$ || (DangerouslyDeleteByTagsTags$ = {}));
export function dangerouslyDeleteByTagsTagsToJSON(dangerouslyDeleteByTagsTags) {
    return JSON.stringify(DangerouslyDeleteByTagsTags$outboundSchema.parse(dangerouslyDeleteByTagsTags));
}
export function dangerouslyDeleteByTagsTagsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DangerouslyDeleteByTagsTags$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DangerouslyDeleteByTagsTags' from JSON`);
}
/** @internal */
export const DangerouslyDeleteByTagsTarget$inboundSchema = z.nativeEnum(DangerouslyDeleteByTagsTarget);
/** @internal */
export const DangerouslyDeleteByTagsTarget$outboundSchema = DangerouslyDeleteByTagsTarget$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DangerouslyDeleteByTagsTarget$;
(function (DangerouslyDeleteByTagsTarget$) {
    /** @deprecated use `DangerouslyDeleteByTagsTarget$inboundSchema` instead. */
    DangerouslyDeleteByTagsTarget$.inboundSchema = DangerouslyDeleteByTagsTarget$inboundSchema;
    /** @deprecated use `DangerouslyDeleteByTagsTarget$outboundSchema` instead. */
    DangerouslyDeleteByTagsTarget$.outboundSchema = DangerouslyDeleteByTagsTarget$outboundSchema;
})(DangerouslyDeleteByTagsTarget$ || (DangerouslyDeleteByTagsTarget$ = {}));
/** @internal */
export const DangerouslyDeleteByTagsRequestBody$inboundSchema = z.object({
    revalidationDeadlineSeconds: z.number().optional(),
    tags: z.union([z.array(z.string()), z.string()]),
    target: DangerouslyDeleteByTagsTarget$inboundSchema.optional(),
});
/** @internal */
export const DangerouslyDeleteByTagsRequestBody$outboundSchema = z.object({
    revalidationDeadlineSeconds: z.number().optional(),
    tags: z.union([z.array(z.string()), z.string()]),
    target: DangerouslyDeleteByTagsTarget$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DangerouslyDeleteByTagsRequestBody$;
(function (DangerouslyDeleteByTagsRequestBody$) {
    /** @deprecated use `DangerouslyDeleteByTagsRequestBody$inboundSchema` instead. */
    DangerouslyDeleteByTagsRequestBody$.inboundSchema = DangerouslyDeleteByTagsRequestBody$inboundSchema;
    /** @deprecated use `DangerouslyDeleteByTagsRequestBody$outboundSchema` instead. */
    DangerouslyDeleteByTagsRequestBody$.outboundSchema = DangerouslyDeleteByTagsRequestBody$outboundSchema;
})(DangerouslyDeleteByTagsRequestBody$ || (DangerouslyDeleteByTagsRequestBody$ = {}));
export function dangerouslyDeleteByTagsRequestBodyToJSON(dangerouslyDeleteByTagsRequestBody) {
    return JSON.stringify(DangerouslyDeleteByTagsRequestBody$outboundSchema.parse(dangerouslyDeleteByTagsRequestBody));
}
export function dangerouslyDeleteByTagsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DangerouslyDeleteByTagsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DangerouslyDeleteByTagsRequestBody' from JSON`);
}
/** @internal */
export const DangerouslyDeleteByTagsRequest$inboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => DangerouslyDeleteByTagsRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const DangerouslyDeleteByTagsRequest$outboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => DangerouslyDeleteByTagsRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DangerouslyDeleteByTagsRequest$;
(function (DangerouslyDeleteByTagsRequest$) {
    /** @deprecated use `DangerouslyDeleteByTagsRequest$inboundSchema` instead. */
    DangerouslyDeleteByTagsRequest$.inboundSchema = DangerouslyDeleteByTagsRequest$inboundSchema;
    /** @deprecated use `DangerouslyDeleteByTagsRequest$outboundSchema` instead. */
    DangerouslyDeleteByTagsRequest$.outboundSchema = DangerouslyDeleteByTagsRequest$outboundSchema;
})(DangerouslyDeleteByTagsRequest$ || (DangerouslyDeleteByTagsRequest$ = {}));
export function dangerouslyDeleteByTagsRequestToJSON(dangerouslyDeleteByTagsRequest) {
    return JSON.stringify(DangerouslyDeleteByTagsRequest$outboundSchema.parse(dangerouslyDeleteByTagsRequest));
}
export function dangerouslyDeleteByTagsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DangerouslyDeleteByTagsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DangerouslyDeleteByTagsRequest' from JSON`);
}
