// @vercel/sdk/esm/models/domainnotregistered.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const DomainNotRegisteredCode = {
    DomainNotRegistered: "domain_not_registered",
};
/**
 * The domain is not registered with Vercel.
 */
export class DomainNotRegistered extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "DomainNotRegistered";
    }
}
/** @internal */
export const DomainNotRegisteredCode$inboundSchema = z.nativeEnum(DomainNotRegisteredCode);
/** @internal */
export const DomainNotRegisteredCode$outboundSchema = DomainNotRegisteredCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DomainNotRegisteredCode$;
(function (DomainNotRegisteredCode$) {
    /** @deprecated use `DomainNotRegisteredCode$inboundSchema` instead. */
    DomainNotRegisteredCode$.inboundSchema = DomainNotRegisteredCode$inboundSchema;
    /** @deprecated use `DomainNotRegisteredCode$outboundSchema` instead. */
    DomainNotRegisteredCode$.outboundSchema = DomainNotRegisteredCode$outboundSchema;
})(DomainNotRegisteredCode$ || (DomainNotRegisteredCode$ = {}));
/** @internal */
export const DomainNotRegistered$inboundSchema = z.object({
    status: z.number(),
    code: DomainNotRegisteredCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new DomainNotRegistered(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const DomainNotRegistered$outboundSchema = z.instanceof(DomainNotRegistered)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: DomainNotRegisteredCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DomainNotRegistered$;
(function (DomainNotRegistered$) {
    /** @deprecated use `DomainNotRegistered$inboundSchema` instead. */
    DomainNotRegistered$.inboundSchema = DomainNotRegistered$inboundSchema;
    /** @deprecated use `DomainNotRegistered$outboundSchema` instead. */
    DomainNotRegistered$.outboundSchema = DomainNotRegistered$outboundSchema;
})(DomainNotRegistered$ || (DomainNotRegistered$ = {}));
