// @vercel/sdk/esm/models/domainnotrenewable.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const DomainNotRenewableCode = {
    DomainNotRenewable: "domain_not_renewable",
};
/**
 * The domain is not renewable.
 */
export class DomainNotRenewable extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "DomainNotRenewable";
    }
}
/** @internal */
export const DomainNotRenewableCode$inboundSchema = z.nativeEnum(DomainNotRenewableCode);
/** @internal */
export const DomainNotRenewableCode$outboundSchema = DomainNotRenewableCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DomainNotRenewableCode$;
(function (DomainNotRenewableCode$) {
    /** @deprecated use `DomainNotRenewableCode$inboundSchema` instead. */
    DomainNotRenewableCode$.inboundSchema = DomainNotRenewableCode$inboundSchema;
    /** @deprecated use `DomainNotRenewableCode$outboundSchema` instead. */
    DomainNotRenewableCode$.outboundSchema = DomainNotRenewableCode$outboundSchema;
})(DomainNotRenewableCode$ || (DomainNotRenewableCode$ = {}));
/** @internal */
export const DomainNotRenewable$inboundSchema = z.object({
    status: z.number(),
    code: DomainNotRenewableCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new DomainNotRenewable(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const DomainNotRenewable$outboundSchema = z.instanceof(DomainNotRenewable)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: DomainNotRenewableCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DomainNotRenewable$;
(function (DomainNotRenewable$) {
    /** @deprecated use `DomainNotRenewable$inboundSchema` instead. */
    DomainNotRenewable$.inboundSchema = DomainNotRenewable$inboundSchema;
    /** @deprecated use `DomainNotRenewable$outboundSchema` instead. */
    DomainNotRenewable$.outboundSchema = DomainNotRenewable$outboundSchema;
})(DomainNotRenewable$ || (DomainNotRenewable$ = {}));
