// @vercel/sdk/esm/models/getbulkavailabilityop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetBulkAvailabilityRequestBody$inboundSchema = z.object({
    domains: z.array(z.string()),
});
/** @internal */
export const GetBulkAvailabilityRequestBody$outboundSchema = z.object({
    domains: z.array(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetBulkAvailabilityRequestBody$;
(function (GetBulkAvailabilityRequestBody$) {
    /** @deprecated use `GetBulkAvailabilityRequestBody$inboundSchema` instead. */
    GetBulkAvailabilityRequestBody$.inboundSchema = GetBulkAvailabilityRequestBody$inboundSchema;
    /** @deprecated use `GetBulkAvailabilityRequestBody$outboundSchema` instead. */
    GetBulkAvailabilityRequestBody$.outboundSchema = GetBulkAvailabilityRequestBody$outboundSchema;
})(GetBulkAvailabilityRequestBody$ || (GetBulkAvailabilityRequestBody$ = {}));
export function getBulkAvailabilityRequestBodyToJSON(getBulkAvailabilityRequestBody) {
    return JSON.stringify(GetBulkAvailabilityRequestBody$outboundSchema.parse(getBulkAvailabilityRequestBody));
}
export function getBulkAvailabilityRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBulkAvailabilityRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBulkAvailabilityRequestBody' from JSON`);
}
/** @internal */
export const GetBulkAvailabilityRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    RequestBody: z.lazy(() => GetBulkAvailabilityRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const GetBulkAvailabilityRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    requestBody: z.lazy(() => GetBulkAvailabilityRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetBulkAvailabilityRequest$;
(function (GetBulkAvailabilityRequest$) {
    /** @deprecated use `GetBulkAvailabilityRequest$inboundSchema` instead. */
    GetBulkAvailabilityRequest$.inboundSchema = GetBulkAvailabilityRequest$inboundSchema;
    /** @deprecated use `GetBulkAvailabilityRequest$outboundSchema` instead. */
    GetBulkAvailabilityRequest$.outboundSchema = GetBulkAvailabilityRequest$outboundSchema;
})(GetBulkAvailabilityRequest$ || (GetBulkAvailabilityRequest$ = {}));
export function getBulkAvailabilityRequestToJSON(getBulkAvailabilityRequest) {
    return JSON.stringify(GetBulkAvailabilityRequest$outboundSchema.parse(getBulkAvailabilityRequest));
}
export function getBulkAvailabilityRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBulkAvailabilityRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBulkAvailabilityRequest' from JSON`);
}
/** @internal */
export const Results$inboundSchema = z.object({
    domain: z.string(),
    available: z.boolean(),
});
/** @internal */
export const Results$outboundSchema = z.object({
    domain: z.string(),
    available: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Results$;
(function (Results$) {
    /** @deprecated use `Results$inboundSchema` instead. */
    Results$.inboundSchema = Results$inboundSchema;
    /** @deprecated use `Results$outboundSchema` instead. */
    Results$.outboundSchema = Results$outboundSchema;
})(Results$ || (Results$ = {}));
export function resultsToJSON(results) {
    return JSON.stringify(Results$outboundSchema.parse(results));
}
export function resultsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Results$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Results' from JSON`);
}
/** @internal */
export const GetBulkAvailabilityResponseBody$inboundSchema = z.object({
    results: z.array(z.lazy(() => Results$inboundSchema)),
});
/** @internal */
export const GetBulkAvailabilityResponseBody$outboundSchema = z.object({
    results: z.array(z.lazy(() => Results$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetBulkAvailabilityResponseBody$;
(function (GetBulkAvailabilityResponseBody$) {
    /** @deprecated use `GetBulkAvailabilityResponseBody$inboundSchema` instead. */
    GetBulkAvailabilityResponseBody$.inboundSchema = GetBulkAvailabilityResponseBody$inboundSchema;
    /** @deprecated use `GetBulkAvailabilityResponseBody$outboundSchema` instead. */
    GetBulkAvailabilityResponseBody$.outboundSchema = GetBulkAvailabilityResponseBody$outboundSchema;
})(GetBulkAvailabilityResponseBody$ || (GetBulkAvailabilityResponseBody$ = {}));
export function getBulkAvailabilityResponseBodyToJSON(getBulkAvailabilityResponseBody) {
    return JSON.stringify(GetBulkAvailabilityResponseBody$outboundSchema.parse(getBulkAvailabilityResponseBody));
}
export function getBulkAvailabilityResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBulkAvailabilityResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBulkAvailabilityResponseBody' from JSON`);
}
