// @vercel/sdk/esm/models/getdomainavailabilityop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetDomainAvailabilityRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetDomainAvailabilityRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainAvailabilityRequest$;
(function (GetDomainAvailabilityRequest$) {
    /** @deprecated use `GetDomainAvailabilityRequest$inboundSchema` instead. */
    GetDomainAvailabilityRequest$.inboundSchema = GetDomainAvailabilityRequest$inboundSchema;
    /** @deprecated use `GetDomainAvailabilityRequest$outboundSchema` instead. */
    GetDomainAvailabilityRequest$.outboundSchema = GetDomainAvailabilityRequest$outboundSchema;
})(GetDomainAvailabilityRequest$ || (GetDomainAvailabilityRequest$ = {}));
export function getDomainAvailabilityRequestToJSON(getDomainAvailabilityRequest) {
    return JSON.stringify(GetDomainAvailabilityRequest$outboundSchema.parse(getDomainAvailabilityRequest));
}
export function getDomainAvailabilityRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainAvailabilityRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainAvailabilityRequest' from JSON`);
}
/** @internal */
export const GetDomainAvailabilityResponseBody$inboundSchema = z.object({
    available: z.boolean(),
});
/** @internal */
export const GetDomainAvailabilityResponseBody$outboundSchema = z.object({
    available: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainAvailabilityResponseBody$;
(function (GetDomainAvailabilityResponseBody$) {
    /** @deprecated use `GetDomainAvailabilityResponseBody$inboundSchema` instead. */
    GetDomainAvailabilityResponseBody$.inboundSchema = GetDomainAvailabilityResponseBody$inboundSchema;
    /** @deprecated use `GetDomainAvailabilityResponseBody$outboundSchema` instead. */
    GetDomainAvailabilityResponseBody$.outboundSchema = GetDomainAvailabilityResponseBody$outboundSchema;
})(GetDomainAvailabilityResponseBody$ || (GetDomainAvailabilityResponseBody$ = {}));
export function getDomainAvailabilityResponseBodyToJSON(getDomainAvailabilityResponseBody) {
    return JSON.stringify(GetDomainAvailabilityResponseBody$outboundSchema.parse(getDomainAvailabilityResponseBody));
}
export function getDomainAvailabilityResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainAvailabilityResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainAvailabilityResponseBody' from JSON`);
}
