// @vercel/sdk/esm/models/getdomaintransferinop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
export const GetDomainTransferInStatus = {
    Canceled: "canceled",
    CanceledPendingRefund: "canceled_pending_refund",
    Completed: "completed",
    Created: "created",
    Failed: "failed",
    Pending: "pending",
    PendingInsert: "pending_insert",
    PendingNewAuthCode: "pending_new_auth_code",
    PendingTransfer: "pending_transfer",
    PendingUnlock: "pending_unlock",
    Rejected: "rejected",
    SubmittingTransfer: "submitting_transfer",
};
/** @internal */
export const GetDomainTransferInRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetDomainTransferInRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainTransferInRequest$;
(function (GetDomainTransferInRequest$) {
    /** @deprecated use `GetDomainTransferInRequest$inboundSchema` instead. */
    GetDomainTransferInRequest$.inboundSchema = GetDomainTransferInRequest$inboundSchema;
    /** @deprecated use `GetDomainTransferInRequest$outboundSchema` instead. */
    GetDomainTransferInRequest$.outboundSchema = GetDomainTransferInRequest$outboundSchema;
})(GetDomainTransferInRequest$ || (GetDomainTransferInRequest$ = {}));
export function getDomainTransferInRequestToJSON(getDomainTransferInRequest) {
    return JSON.stringify(GetDomainTransferInRequest$outboundSchema.parse(getDomainTransferInRequest));
}
export function getDomainTransferInRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferInRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferInRequest' from JSON`);
}
/** @internal */
export const GetDomainTransferInDomainsRegistrarResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const GetDomainTransferInDomainsRegistrarResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainTransferInDomainsRegistrarResponseBody$;
(function (GetDomainTransferInDomainsRegistrarResponseBody$) {
    /** @deprecated use `GetDomainTransferInDomainsRegistrarResponseBody$inboundSchema` instead. */
    GetDomainTransferInDomainsRegistrarResponseBody$.inboundSchema = GetDomainTransferInDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `GetDomainTransferInDomainsRegistrarResponseBody$outboundSchema` instead. */
    GetDomainTransferInDomainsRegistrarResponseBody$.outboundSchema = GetDomainTransferInDomainsRegistrarResponseBody$outboundSchema;
})(GetDomainTransferInDomainsRegistrarResponseBody$ || (GetDomainTransferInDomainsRegistrarResponseBody$ = {}));
export function getDomainTransferInDomainsRegistrarResponseBodyToJSON(getDomainTransferInDomainsRegistrarResponseBody) {
    return JSON.stringify(GetDomainTransferInDomainsRegistrarResponseBody$outboundSchema.parse(getDomainTransferInDomainsRegistrarResponseBody));
}
export function getDomainTransferInDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferInDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferInDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetDomainTransferInStatus$inboundSchema = z.nativeEnum(GetDomainTransferInStatus);
/** @internal */
export const GetDomainTransferInStatus$outboundSchema = GetDomainTransferInStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainTransferInStatus$;
(function (GetDomainTransferInStatus$) {
    /** @deprecated use `GetDomainTransferInStatus$inboundSchema` instead. */
    GetDomainTransferInStatus$.inboundSchema = GetDomainTransferInStatus$inboundSchema;
    /** @deprecated use `GetDomainTransferInStatus$outboundSchema` instead. */
    GetDomainTransferInStatus$.outboundSchema = GetDomainTransferInStatus$outboundSchema;
})(GetDomainTransferInStatus$ || (GetDomainTransferInStatus$ = {}));
/** @internal */
export const GetDomainTransferInResponseBody$inboundSchema = z.object({
    status: GetDomainTransferInStatus$inboundSchema,
});
/** @internal */
export const GetDomainTransferInResponseBody$outboundSchema = z.object({
    status: GetDomainTransferInStatus$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainTransferInResponseBody$;
(function (GetDomainTransferInResponseBody$) {
    /** @deprecated use `GetDomainTransferInResponseBody$inboundSchema` instead. */
    GetDomainTransferInResponseBody$.inboundSchema = GetDomainTransferInResponseBody$inboundSchema;
    /** @deprecated use `GetDomainTransferInResponseBody$outboundSchema` instead. */
    GetDomainTransferInResponseBody$.outboundSchema = GetDomainTransferInResponseBody$outboundSchema;
})(GetDomainTransferInResponseBody$ || (GetDomainTransferInResponseBody$ = {}));
export function getDomainTransferInResponseBodyToJSON(getDomainTransferInResponseBody) {
    return JSON.stringify(GetDomainTransferInResponseBody$outboundSchema.parse(getDomainTransferInResponseBody));
}
export function getDomainTransferInResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferInResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferInResponseBody' from JSON`);
}
