// @vercel/sdk/esm/models/getdomainauthcodeop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { DomainNotRegistered$inboundSchema, DomainNotRegistered$outboundSchema, } from "./domainnotregistered.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
/** @internal */
export const GetDomainAuthCodeRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetDomainAuthCodeRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainAuthCodeRequest$;
(function (GetDomainAuthCodeRequest$) {
    /** @deprecated use `GetDomainAuthCodeRequest$inboundSchema` instead. */
    GetDomainAuthCodeRequest$.inboundSchema = GetDomainAuthCodeRequest$inboundSchema;
    /** @deprecated use `GetDomainAuthCodeRequest$outboundSchema` instead. */
    GetDomainAuthCodeRequest$.outboundSchema = GetDomainAuthCodeRequest$outboundSchema;
})(GetDomainAuthCodeRequest$ || (GetDomainAuthCodeRequest$ = {}));
export function getDomainAuthCodeRequestToJSON(getDomainAuthCodeRequest) {
    return JSON.stringify(GetDomainAuthCodeRequest$outboundSchema.parse(getDomainAuthCodeRequest));
}
export function getDomainAuthCodeRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainAuthCodeRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainAuthCodeRequest' from JSON`);
}
/** @internal */
export const GetDomainAuthCodeDomainsRegistrarResponseResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const GetDomainAuthCodeDomainsRegistrarResponseResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainAuthCodeDomainsRegistrarResponseResponseBody$;
(function (GetDomainAuthCodeDomainsRegistrarResponseResponseBody$) {
    /** @deprecated use `GetDomainAuthCodeDomainsRegistrarResponseResponseBody$inboundSchema` instead. */
    GetDomainAuthCodeDomainsRegistrarResponseResponseBody$.inboundSchema = GetDomainAuthCodeDomainsRegistrarResponseResponseBody$inboundSchema;
    /** @deprecated use `GetDomainAuthCodeDomainsRegistrarResponseResponseBody$outboundSchema` instead. */
    GetDomainAuthCodeDomainsRegistrarResponseResponseBody$.outboundSchema = GetDomainAuthCodeDomainsRegistrarResponseResponseBody$outboundSchema;
})(GetDomainAuthCodeDomainsRegistrarResponseResponseBody$ || (GetDomainAuthCodeDomainsRegistrarResponseResponseBody$ = {}));
export function getDomainAuthCodeDomainsRegistrarResponseResponseBodyToJSON(getDomainAuthCodeDomainsRegistrarResponseResponseBody) {
    return JSON.stringify(GetDomainAuthCodeDomainsRegistrarResponseResponseBody$outboundSchema.parse(getDomainAuthCodeDomainsRegistrarResponseResponseBody));
}
export function getDomainAuthCodeDomainsRegistrarResponseResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainAuthCodeDomainsRegistrarResponseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainAuthCodeDomainsRegistrarResponseResponseBody' from JSON`);
}
/** @internal */
export const GetDomainAuthCodeDomainsRegistrarResponseBody$inboundSchema = z.union([
    DomainNotRegistered$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const GetDomainAuthCodeDomainsRegistrarResponseBody$outboundSchema = z.union([
    DomainNotRegistered$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainAuthCodeDomainsRegistrarResponseBody$;
(function (GetDomainAuthCodeDomainsRegistrarResponseBody$) {
    /** @deprecated use `GetDomainAuthCodeDomainsRegistrarResponseBody$inboundSchema` instead. */
    GetDomainAuthCodeDomainsRegistrarResponseBody$.inboundSchema = GetDomainAuthCodeDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `GetDomainAuthCodeDomainsRegistrarResponseBody$outboundSchema` instead. */
    GetDomainAuthCodeDomainsRegistrarResponseBody$.outboundSchema = GetDomainAuthCodeDomainsRegistrarResponseBody$outboundSchema;
})(GetDomainAuthCodeDomainsRegistrarResponseBody$ || (GetDomainAuthCodeDomainsRegistrarResponseBody$ = {}));
export function getDomainAuthCodeDomainsRegistrarResponseBodyToJSON(getDomainAuthCodeDomainsRegistrarResponseBody) {
    return JSON.stringify(GetDomainAuthCodeDomainsRegistrarResponseBody$outboundSchema.parse(getDomainAuthCodeDomainsRegistrarResponseBody));
}
export function getDomainAuthCodeDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainAuthCodeDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainAuthCodeDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetDomainAuthCodeResponseBody$inboundSchema = z.object({
    authCode: z.string(),
});
/** @internal */
export const GetDomainAuthCodeResponseBody$outboundSchema = z.object({
    authCode: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetDomainAuthCodeResponseBody$;
(function (GetDomainAuthCodeResponseBody$) {
    /** @deprecated use `GetDomainAuthCodeResponseBody$inboundSchema` instead. */
    GetDomainAuthCodeResponseBody$.inboundSchema = GetDomainAuthCodeResponseBody$inboundSchema;
    /** @deprecated use `GetDomainAuthCodeResponseBody$outboundSchema` instead. */
    GetDomainAuthCodeResponseBody$.outboundSchema = GetDomainAuthCodeResponseBody$outboundSchema;
})(GetDomainAuthCodeResponseBody$ || (GetDomainAuthCodeResponseBody$ = {}));
export function getDomainAuthCodeResponseBodyToJSON(getDomainAuthCodeResponseBody) {
    return JSON.stringify(GetDomainAuthCodeResponseBody$outboundSchema.parse(getDomainAuthCodeResponseBody));
}
export function getDomainAuthCodeResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainAuthCodeResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainAuthCodeResponseBody' from JSON`);
}
