// @vercel/sdk/esm/models/getintegrationlogdrainsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/**
 * The delivery log format
 */
export const GetIntegrationLogDrainsDeliveryFormat = {
    Json: "json",
    Ndjson: "ndjson",
    Syslog: "syslog",
    Protobuf: "protobuf",
};
/**
 * The sources from which logs are currently being delivered to this log drain.
 */
export const GetIntegrationLogDrainsSources = {
    Build: "build",
    Edge: "edge",
    Lambda: "lambda",
    Static: "static",
    External: "external",
    Firewall: "firewall",
};
/**
 * Whether the log drain was created by an integration or by a user
 */
export const CreatedFrom = {
    Integration: "integration",
    SelfServed: "self-served",
};
/**
 * The environment of log drain
 */
export const GetIntegrationLogDrainsEnvironments = {
    Production: "production",
    Preview: "preview",
};
export const SourceKind = {
    Integration: "integration",
};
export const Kind = {
    SelfServed: "self-served",
};
/** @internal */
export const GetIntegrationLogDrainsRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetIntegrationLogDrainsRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetIntegrationLogDrainsRequest$;
(function (GetIntegrationLogDrainsRequest$) {
    /** @deprecated use `GetIntegrationLogDrainsRequest$inboundSchema` instead. */
    GetIntegrationLogDrainsRequest$.inboundSchema = GetIntegrationLogDrainsRequest$inboundSchema;
    /** @deprecated use `GetIntegrationLogDrainsRequest$outboundSchema` instead. */
    GetIntegrationLogDrainsRequest$.outboundSchema = GetIntegrationLogDrainsRequest$outboundSchema;
})(GetIntegrationLogDrainsRequest$ || (GetIntegrationLogDrainsRequest$ = {}));
export function getIntegrationLogDrainsRequestToJSON(getIntegrationLogDrainsRequest) {
    return JSON.stringify(GetIntegrationLogDrainsRequest$outboundSchema.parse(getIntegrationLogDrainsRequest));
}
export function getIntegrationLogDrainsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsRequest' from JSON`);
}
/** @internal */
export const GetIntegrationLogDrainsDeliveryFormat$inboundSchema = z.nativeEnum(GetIntegrationLogDrainsDeliveryFormat);
/** @internal */
export const GetIntegrationLogDrainsDeliveryFormat$outboundSchema = GetIntegrationLogDrainsDeliveryFormat$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetIntegrationLogDrainsDeliveryFormat$;
(function (GetIntegrationLogDrainsDeliveryFormat$) {
    /** @deprecated use `GetIntegrationLogDrainsDeliveryFormat$inboundSchema` instead. */
    GetIntegrationLogDrainsDeliveryFormat$.inboundSchema = GetIntegrationLogDrainsDeliveryFormat$inboundSchema;
    /** @deprecated use `GetIntegrationLogDrainsDeliveryFormat$outboundSchema` instead. */
    GetIntegrationLogDrainsDeliveryFormat$.outboundSchema = GetIntegrationLogDrainsDeliveryFormat$outboundSchema;
})(GetIntegrationLogDrainsDeliveryFormat$ || (GetIntegrationLogDrainsDeliveryFormat$ = {}));
/** @internal */
export const GetIntegrationLogDrainsSources$inboundSchema = z.nativeEnum(GetIntegrationLogDrainsSources);
/** @internal */
export const GetIntegrationLogDrainsSources$outboundSchema = GetIntegrationLogDrainsSources$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetIntegrationLogDrainsSources$;
(function (GetIntegrationLogDrainsSources$) {
    /** @deprecated use `GetIntegrationLogDrainsSources$inboundSchema` instead. */
    GetIntegrationLogDrainsSources$.inboundSchema = GetIntegrationLogDrainsSources$inboundSchema;
    /** @deprecated use `GetIntegrationLogDrainsSources$outboundSchema` instead. */
    GetIntegrationLogDrainsSources$.outboundSchema = GetIntegrationLogDrainsSources$outboundSchema;
})(GetIntegrationLogDrainsSources$ || (GetIntegrationLogDrainsSources$ = {}));
/** @internal */
export const CreatedFrom$inboundSchema = z
    .nativeEnum(CreatedFrom);
/** @internal */
export const CreatedFrom$outboundSchema = CreatedFrom$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CreatedFrom$;
(function (CreatedFrom$) {
    /** @deprecated use `CreatedFrom$inboundSchema` instead. */
    CreatedFrom$.inboundSchema = CreatedFrom$inboundSchema;
    /** @deprecated use `CreatedFrom$outboundSchema` instead. */
    CreatedFrom$.outboundSchema = CreatedFrom$outboundSchema;
})(CreatedFrom$ || (CreatedFrom$ = {}));
/** @internal */
export const GetIntegrationLogDrainsEnvironments$inboundSchema = z.nativeEnum(GetIntegrationLogDrainsEnvironments);
/** @internal */
export const GetIntegrationLogDrainsEnvironments$outboundSchema = GetIntegrationLogDrainsEnvironments$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetIntegrationLogDrainsEnvironments$;
(function (GetIntegrationLogDrainsEnvironments$) {
    /** @deprecated use `GetIntegrationLogDrainsEnvironments$inboundSchema` instead. */
    GetIntegrationLogDrainsEnvironments$.inboundSchema = GetIntegrationLogDrainsEnvironments$inboundSchema;
    /** @deprecated use `GetIntegrationLogDrainsEnvironments$outboundSchema` instead. */
    GetIntegrationLogDrainsEnvironments$.outboundSchema = GetIntegrationLogDrainsEnvironments$outboundSchema;
})(GetIntegrationLogDrainsEnvironments$ || (GetIntegrationLogDrainsEnvironments$ = {}));
/** @internal */
export const SourceKind$inboundSchema = z
    .nativeEnum(SourceKind);
/** @internal */
export const SourceKind$outboundSchema = SourceKind$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SourceKind$;
(function (SourceKind$) {
    /** @deprecated use `SourceKind$inboundSchema` instead. */
    SourceKind$.inboundSchema = SourceKind$inboundSchema;
    /** @deprecated use `SourceKind$outboundSchema` instead. */
    SourceKind$.outboundSchema = SourceKind$outboundSchema;
})(SourceKind$ || (SourceKind$ = {}));
/** @internal */
export const Source2$inboundSchema = z.object({
    kind: SourceKind$inboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const Source2$outboundSchema = z.object({
    kind: SourceKind$outboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Source2$;
(function (Source2$) {
    /** @deprecated use `Source2$inboundSchema` instead. */
    Source2$.inboundSchema = Source2$inboundSchema;
    /** @deprecated use `Source2$outboundSchema` instead. */
    Source2$.outboundSchema = Source2$outboundSchema;
})(Source2$ || (Source2$ = {}));
export function source2ToJSON(source2) {
    return JSON.stringify(Source2$outboundSchema.parse(source2));
}
export function source2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Source2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Source2' from JSON`);
}
/** @internal */
export const Kind$inboundSchema = z.nativeEnum(Kind);
/** @internal */
export const Kind$outboundSchema = Kind$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Kind$;
(function (Kind$) {
    /** @deprecated use `Kind$inboundSchema` instead. */
    Kind$.inboundSchema = Kind$inboundSchema;
    /** @deprecated use `Kind$outboundSchema` instead. */
    Kind$.outboundSchema = Kind$outboundSchema;
})(Kind$ || (Kind$ = {}));
/** @internal */
export const Source1$inboundSchema = z.object({
    kind: Kind$inboundSchema,
});
/** @internal */
export const Source1$outboundSchema = z.object({
    kind: Kind$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Source1$;
(function (Source1$) {
    /** @deprecated use `Source1$inboundSchema` instead. */
    Source1$.inboundSchema = Source1$inboundSchema;
    /** @deprecated use `Source1$outboundSchema` instead. */
    Source1$.outboundSchema = Source1$outboundSchema;
})(Source1$ || (Source1$ = {}));
export function source1ToJSON(source1) {
    return JSON.stringify(Source1$outboundSchema.parse(source1));
}
export function source1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Source1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Source1' from JSON`);
}
/** @internal */
export const GetIntegrationLogDrainsSource$inboundSchema = z.union([
    z.lazy(() => Source2$inboundSchema),
    z.lazy(() => Source1$inboundSchema),
]);
/** @internal */
export const GetIntegrationLogDrainsSource$outboundSchema = z.union([
    z.lazy(() => Source2$outboundSchema),
    z.lazy(() => Source1$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetIntegrationLogDrainsSource$;
(function (GetIntegrationLogDrainsSource$) {
    /** @deprecated use `GetIntegrationLogDrainsSource$inboundSchema` instead. */
    GetIntegrationLogDrainsSource$.inboundSchema = GetIntegrationLogDrainsSource$inboundSchema;
    /** @deprecated use `GetIntegrationLogDrainsSource$outboundSchema` instead. */
    GetIntegrationLogDrainsSource$.outboundSchema = GetIntegrationLogDrainsSource$outboundSchema;
})(GetIntegrationLogDrainsSource$ || (GetIntegrationLogDrainsSource$ = {}));
export function getIntegrationLogDrainsSourceToJSON(getIntegrationLogDrainsSource) {
    return JSON.stringify(GetIntegrationLogDrainsSource$outboundSchema.parse(getIntegrationLogDrainsSource));
}
export function getIntegrationLogDrainsSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsSource' from JSON`);
}
/** @internal */
export const GetIntegrationLogDrainsResponseBody$inboundSchema = z.object({
    clientId: z.string().optional(),
    configurationId: z.string().optional(),
    createdAt: z.number(),
    id: z.string(),
    deliveryFormat: GetIntegrationLogDrainsDeliveryFormat$inboundSchema
        .optional(),
    name: z.string(),
    ownerId: z.string(),
    projectId: z.nullable(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    url: z.string(),
    sources: z.array(GetIntegrationLogDrainsSources$inboundSchema).optional(),
    createdFrom: CreatedFrom$inboundSchema.optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(GetIntegrationLogDrainsEnvironments$inboundSchema)
        .optional(),
    branch: z.string().optional(),
    samplingRate: z.number().optional(),
    source: z.union([
        z.lazy(() => Source2$inboundSchema),
        z.lazy(() => Source1$inboundSchema),
    ]),
});
/** @internal */
export const GetIntegrationLogDrainsResponseBody$outboundSchema = z.object({
    clientId: z.string().optional(),
    configurationId: z.string().optional(),
    createdAt: z.number(),
    id: z.string(),
    deliveryFormat: GetIntegrationLogDrainsDeliveryFormat$outboundSchema
        .optional(),
    name: z.string(),
    ownerId: z.string(),
    projectId: z.nullable(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    url: z.string(),
    sources: z.array(GetIntegrationLogDrainsSources$outboundSchema).optional(),
    createdFrom: CreatedFrom$outboundSchema.optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(GetIntegrationLogDrainsEnvironments$outboundSchema)
        .optional(),
    branch: z.string().optional(),
    samplingRate: z.number().optional(),
    source: z.union([
        z.lazy(() => Source2$outboundSchema),
        z.lazy(() => Source1$outboundSchema),
    ]),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetIntegrationLogDrainsResponseBody$;
(function (GetIntegrationLogDrainsResponseBody$) {
    /** @deprecated use `GetIntegrationLogDrainsResponseBody$inboundSchema` instead. */
    GetIntegrationLogDrainsResponseBody$.inboundSchema = GetIntegrationLogDrainsResponseBody$inboundSchema;
    /** @deprecated use `GetIntegrationLogDrainsResponseBody$outboundSchema` instead. */
    GetIntegrationLogDrainsResponseBody$.outboundSchema = GetIntegrationLogDrainsResponseBody$outboundSchema;
})(GetIntegrationLogDrainsResponseBody$ || (GetIntegrationLogDrainsResponseBody$ = {}));
export function getIntegrationLogDrainsResponseBodyToJSON(getIntegrationLogDrainsResponseBody) {
    return JSON.stringify(GetIntegrationLogDrainsResponseBody$outboundSchema.parse(getIntegrationLogDrainsResponseBody));
}
export function getIntegrationLogDrainsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsResponseBody' from JSON`);
}
