// @vercel/sdk/esm/models/getorderop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
export const GetOrderDomainsPurchaseType = {
    Transfer: "transfer",
};
export const GetOrderDomainsDomainsRegistrarStatus = {
    Pending: "pending",
    Completed: "completed",
    Failed: "failed",
    Refunded: "refunded",
    RefundFailed: "refund-failed",
};
export const DomainsPurchaseType = {
    Renewal: "renewal",
};
export const GetOrderDomainsStatus = {
    Pending: "pending",
    Completed: "completed",
    Failed: "failed",
    Refunded: "refunded",
    RefundFailed: "refund-failed",
};
export const GetOrderDomainsDomainsRegistrarPurchaseType = {
    Purchase: "purchase",
};
export const DomainsStatus = {
    Pending: "pending",
    Completed: "completed",
    Failed: "failed",
    Refunded: "refunded",
    RefundFailed: "refund-failed",
};
export const GetOrderStatus = {
    Draft: "draft",
    Purchasing: "purchasing",
    Completed: "completed",
    Failed: "failed",
};
export const GetOrderErrorDomainsRegistrarResponseCode = {
    UnexpectedError: "unexpected-error",
};
export const GetOrderErrorDomainsRegistrarCode = {
    PriceMismatch: "price-mismatch",
};
export const GetOrderErrorCode = {
    TldOutage: "tld-outage",
};
export const ErrorCode = {
    PaymentFailed: "payment-failed",
};
/** @internal */
export const GetOrderRequest$inboundSchema = z.object({
    orderId: z.string(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetOrderRequest$outboundSchema = z.object({
    orderId: z.string(),
    teamId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderRequest$;
(function (GetOrderRequest$) {
    /** @deprecated use `GetOrderRequest$inboundSchema` instead. */
    GetOrderRequest$.inboundSchema = GetOrderRequest$inboundSchema;
    /** @deprecated use `GetOrderRequest$outboundSchema` instead. */
    GetOrderRequest$.outboundSchema = GetOrderRequest$outboundSchema;
})(GetOrderRequest$ || (GetOrderRequest$ = {}));
export function getOrderRequestToJSON(getOrderRequest) {
    return JSON.stringify(GetOrderRequest$outboundSchema.parse(getOrderRequest));
}
export function getOrderRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderRequest' from JSON`);
}
/** @internal */
export const GetOrderDomainsRegistrarResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const GetOrderDomainsRegistrarResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderDomainsRegistrarResponseBody$;
(function (GetOrderDomainsRegistrarResponseBody$) {
    /** @deprecated use `GetOrderDomainsRegistrarResponseBody$inboundSchema` instead. */
    GetOrderDomainsRegistrarResponseBody$.inboundSchema = GetOrderDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `GetOrderDomainsRegistrarResponseBody$outboundSchema` instead. */
    GetOrderDomainsRegistrarResponseBody$.outboundSchema = GetOrderDomainsRegistrarResponseBody$outboundSchema;
})(GetOrderDomainsRegistrarResponseBody$ || (GetOrderDomainsRegistrarResponseBody$ = {}));
export function getOrderDomainsRegistrarResponseBodyToJSON(getOrderDomainsRegistrarResponseBody) {
    return JSON.stringify(GetOrderDomainsRegistrarResponseBody$outboundSchema.parse(getOrderDomainsRegistrarResponseBody));
}
export function getOrderDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetOrderDomainsPurchaseType$inboundSchema = z.nativeEnum(GetOrderDomainsPurchaseType);
/** @internal */
export const GetOrderDomainsPurchaseType$outboundSchema = GetOrderDomainsPurchaseType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderDomainsPurchaseType$;
(function (GetOrderDomainsPurchaseType$) {
    /** @deprecated use `GetOrderDomainsPurchaseType$inboundSchema` instead. */
    GetOrderDomainsPurchaseType$.inboundSchema = GetOrderDomainsPurchaseType$inboundSchema;
    /** @deprecated use `GetOrderDomainsPurchaseType$outboundSchema` instead. */
    GetOrderDomainsPurchaseType$.outboundSchema = GetOrderDomainsPurchaseType$outboundSchema;
})(GetOrderDomainsPurchaseType$ || (GetOrderDomainsPurchaseType$ = {}));
/** @internal */
export const GetOrderDomainsDomainsRegistrarStatus$inboundSchema = z.nativeEnum(GetOrderDomainsDomainsRegistrarStatus);
/** @internal */
export const GetOrderDomainsDomainsRegistrarStatus$outboundSchema = GetOrderDomainsDomainsRegistrarStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderDomainsDomainsRegistrarStatus$;
(function (GetOrderDomainsDomainsRegistrarStatus$) {
    /** @deprecated use `GetOrderDomainsDomainsRegistrarStatus$inboundSchema` instead. */
    GetOrderDomainsDomainsRegistrarStatus$.inboundSchema = GetOrderDomainsDomainsRegistrarStatus$inboundSchema;
    /** @deprecated use `GetOrderDomainsDomainsRegistrarStatus$outboundSchema` instead. */
    GetOrderDomainsDomainsRegistrarStatus$.outboundSchema = GetOrderDomainsDomainsRegistrarStatus$outboundSchema;
})(GetOrderDomainsDomainsRegistrarStatus$ || (GetOrderDomainsDomainsRegistrarStatus$ = {}));
/** @internal */
export const Domains3$inboundSchema = z.object({
    purchaseType: GetOrderDomainsPurchaseType$inboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsDomainsRegistrarStatus$inboundSchema,
    price: z.number(),
});
/** @internal */
export const Domains3$outboundSchema = z.object({
    purchaseType: GetOrderDomainsPurchaseType$outboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsDomainsRegistrarStatus$outboundSchema,
    price: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Domains3$;
(function (Domains3$) {
    /** @deprecated use `Domains3$inboundSchema` instead. */
    Domains3$.inboundSchema = Domains3$inboundSchema;
    /** @deprecated use `Domains3$outboundSchema` instead. */
    Domains3$.outboundSchema = Domains3$outboundSchema;
})(Domains3$ || (Domains3$ = {}));
export function domains3ToJSON(domains3) {
    return JSON.stringify(Domains3$outboundSchema.parse(domains3));
}
export function domains3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domains3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domains3' from JSON`);
}
/** @internal */
export const DomainsPurchaseType$inboundSchema = z.nativeEnum(DomainsPurchaseType);
/** @internal */
export const DomainsPurchaseType$outboundSchema = DomainsPurchaseType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DomainsPurchaseType$;
(function (DomainsPurchaseType$) {
    /** @deprecated use `DomainsPurchaseType$inboundSchema` instead. */
    DomainsPurchaseType$.inboundSchema = DomainsPurchaseType$inboundSchema;
    /** @deprecated use `DomainsPurchaseType$outboundSchema` instead. */
    DomainsPurchaseType$.outboundSchema = DomainsPurchaseType$outboundSchema;
})(DomainsPurchaseType$ || (DomainsPurchaseType$ = {}));
/** @internal */
export const GetOrderDomainsStatus$inboundSchema = z.nativeEnum(GetOrderDomainsStatus);
/** @internal */
export const GetOrderDomainsStatus$outboundSchema = GetOrderDomainsStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderDomainsStatus$;
(function (GetOrderDomainsStatus$) {
    /** @deprecated use `GetOrderDomainsStatus$inboundSchema` instead. */
    GetOrderDomainsStatus$.inboundSchema = GetOrderDomainsStatus$inboundSchema;
    /** @deprecated use `GetOrderDomainsStatus$outboundSchema` instead. */
    GetOrderDomainsStatus$.outboundSchema = GetOrderDomainsStatus$outboundSchema;
})(GetOrderDomainsStatus$ || (GetOrderDomainsStatus$ = {}));
/** @internal */
export const Domains2$inboundSchema = z.object({
    purchaseType: DomainsPurchaseType$inboundSchema,
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsStatus$inboundSchema,
    price: z.number(),
});
/** @internal */
export const Domains2$outboundSchema = z.object({
    purchaseType: DomainsPurchaseType$outboundSchema,
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsStatus$outboundSchema,
    price: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Domains2$;
(function (Domains2$) {
    /** @deprecated use `Domains2$inboundSchema` instead. */
    Domains2$.inboundSchema = Domains2$inboundSchema;
    /** @deprecated use `Domains2$outboundSchema` instead. */
    Domains2$.outboundSchema = Domains2$outboundSchema;
})(Domains2$ || (Domains2$ = {}));
export function domains2ToJSON(domains2) {
    return JSON.stringify(Domains2$outboundSchema.parse(domains2));
}
export function domains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domains2' from JSON`);
}
/** @internal */
export const GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema = z
    .nativeEnum(GetOrderDomainsDomainsRegistrarPurchaseType);
/** @internal */
export const GetOrderDomainsDomainsRegistrarPurchaseType$outboundSchema = GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderDomainsDomainsRegistrarPurchaseType$;
(function (GetOrderDomainsDomainsRegistrarPurchaseType$) {
    /** @deprecated use `GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema` instead. */
    GetOrderDomainsDomainsRegistrarPurchaseType$.inboundSchema = GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema;
    /** @deprecated use `GetOrderDomainsDomainsRegistrarPurchaseType$outboundSchema` instead. */
    GetOrderDomainsDomainsRegistrarPurchaseType$.outboundSchema = GetOrderDomainsDomainsRegistrarPurchaseType$outboundSchema;
})(GetOrderDomainsDomainsRegistrarPurchaseType$ || (GetOrderDomainsDomainsRegistrarPurchaseType$ = {}));
/** @internal */
export const DomainsStatus$inboundSchema = z.nativeEnum(DomainsStatus);
/** @internal */
export const DomainsStatus$outboundSchema = DomainsStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DomainsStatus$;
(function (DomainsStatus$) {
    /** @deprecated use `DomainsStatus$inboundSchema` instead. */
    DomainsStatus$.inboundSchema = DomainsStatus$inboundSchema;
    /** @deprecated use `DomainsStatus$outboundSchema` instead. */
    DomainsStatus$.outboundSchema = DomainsStatus$outboundSchema;
})(DomainsStatus$ || (DomainsStatus$ = {}));
/** @internal */
export const Domains1$inboundSchema = z.object({
    purchaseType: GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: DomainsStatus$inboundSchema,
    price: z.number(),
});
/** @internal */
export const Domains1$outboundSchema = z.object({
    purchaseType: GetOrderDomainsDomainsRegistrarPurchaseType$outboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: DomainsStatus$outboundSchema,
    price: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Domains1$;
(function (Domains1$) {
    /** @deprecated use `Domains1$inboundSchema` instead. */
    Domains1$.inboundSchema = Domains1$inboundSchema;
    /** @deprecated use `Domains1$outboundSchema` instead. */
    Domains1$.outboundSchema = Domains1$outboundSchema;
})(Domains1$ || (Domains1$ = {}));
export function domains1ToJSON(domains1) {
    return JSON.stringify(Domains1$outboundSchema.parse(domains1));
}
export function domains1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domains1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domains1' from JSON`);
}
/** @internal */
export const GetOrderDomains$inboundSchema = z.union([
    z.lazy(() => Domains1$inboundSchema),
    z.lazy(() => Domains3$inboundSchema),
    z.lazy(() => Domains2$inboundSchema),
]);
/** @internal */
export const GetOrderDomains$outboundSchema = z.union([
    z.lazy(() => Domains1$outboundSchema),
    z.lazy(() => Domains3$outboundSchema),
    z.lazy(() => Domains2$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderDomains$;
(function (GetOrderDomains$) {
    /** @deprecated use `GetOrderDomains$inboundSchema` instead. */
    GetOrderDomains$.inboundSchema = GetOrderDomains$inboundSchema;
    /** @deprecated use `GetOrderDomains$outboundSchema` instead. */
    GetOrderDomains$.outboundSchema = GetOrderDomains$outboundSchema;
})(GetOrderDomains$ || (GetOrderDomains$ = {}));
export function getOrderDomainsToJSON(getOrderDomains) {
    return JSON.stringify(GetOrderDomains$outboundSchema.parse(getOrderDomains));
}
export function getOrderDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderDomains' from JSON`);
}
/** @internal */
export const GetOrderStatus$inboundSchema = z.nativeEnum(GetOrderStatus);
/** @internal */
export const GetOrderStatus$outboundSchema = GetOrderStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderStatus$;
(function (GetOrderStatus$) {
    /** @deprecated use `GetOrderStatus$inboundSchema` instead. */
    GetOrderStatus$.inboundSchema = GetOrderStatus$inboundSchema;
    /** @deprecated use `GetOrderStatus$outboundSchema` instead. */
    GetOrderStatus$.outboundSchema = GetOrderStatus$outboundSchema;
})(GetOrderStatus$ || (GetOrderStatus$ = {}));
/** @internal */
export const GetOrderErrorDomainsRegistrarResponseCode$inboundSchema = z
    .nativeEnum(GetOrderErrorDomainsRegistrarResponseCode);
/** @internal */
export const GetOrderErrorDomainsRegistrarResponseCode$outboundSchema = GetOrderErrorDomainsRegistrarResponseCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderErrorDomainsRegistrarResponseCode$;
(function (GetOrderErrorDomainsRegistrarResponseCode$) {
    /** @deprecated use `GetOrderErrorDomainsRegistrarResponseCode$inboundSchema` instead. */
    GetOrderErrorDomainsRegistrarResponseCode$.inboundSchema = GetOrderErrorDomainsRegistrarResponseCode$inboundSchema;
    /** @deprecated use `GetOrderErrorDomainsRegistrarResponseCode$outboundSchema` instead. */
    GetOrderErrorDomainsRegistrarResponseCode$.outboundSchema = GetOrderErrorDomainsRegistrarResponseCode$outboundSchema;
})(GetOrderErrorDomainsRegistrarResponseCode$ || (GetOrderErrorDomainsRegistrarResponseCode$ = {}));
/** @internal */
export const Error4$inboundSchema = z
    .object({
    code: GetOrderErrorDomainsRegistrarResponseCode$inboundSchema,
});
/** @internal */
export const Error4$outboundSchema = z.object({
    code: GetOrderErrorDomainsRegistrarResponseCode$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Error4$;
(function (Error4$) {
    /** @deprecated use `Error4$inboundSchema` instead. */
    Error4$.inboundSchema = Error4$inboundSchema;
    /** @deprecated use `Error4$outboundSchema` instead. */
    Error4$.outboundSchema = Error4$outboundSchema;
})(Error4$ || (Error4$ = {}));
export function error4ToJSON(error4) {
    return JSON.stringify(Error4$outboundSchema.parse(error4));
}
export function error4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Error4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Error4' from JSON`);
}
/** @internal */
export const GetOrderErrorDomainsRegistrarCode$inboundSchema = z.nativeEnum(GetOrderErrorDomainsRegistrarCode);
/** @internal */
export const GetOrderErrorDomainsRegistrarCode$outboundSchema = GetOrderErrorDomainsRegistrarCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderErrorDomainsRegistrarCode$;
(function (GetOrderErrorDomainsRegistrarCode$) {
    /** @deprecated use `GetOrderErrorDomainsRegistrarCode$inboundSchema` instead. */
    GetOrderErrorDomainsRegistrarCode$.inboundSchema = GetOrderErrorDomainsRegistrarCode$inboundSchema;
    /** @deprecated use `GetOrderErrorDomainsRegistrarCode$outboundSchema` instead. */
    GetOrderErrorDomainsRegistrarCode$.outboundSchema = GetOrderErrorDomainsRegistrarCode$outboundSchema;
})(GetOrderErrorDomainsRegistrarCode$ || (GetOrderErrorDomainsRegistrarCode$ = {}));
/** @internal */
export const ErrorDetails$inboundSchema = z.object({
    expectedPrice: z.number(),
    actualPrice: z.number().optional(),
});
/** @internal */
export const ErrorDetails$outboundSchema = z.object({
    expectedPrice: z.number(),
    actualPrice: z.number().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ErrorDetails$;
(function (ErrorDetails$) {
    /** @deprecated use `ErrorDetails$inboundSchema` instead. */
    ErrorDetails$.inboundSchema = ErrorDetails$inboundSchema;
    /** @deprecated use `ErrorDetails$outboundSchema` instead. */
    ErrorDetails$.outboundSchema = ErrorDetails$outboundSchema;
})(ErrorDetails$ || (ErrorDetails$ = {}));
export function errorDetailsToJSON(errorDetails) {
    return JSON.stringify(ErrorDetails$outboundSchema.parse(errorDetails));
}
export function errorDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ErrorDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ErrorDetails' from JSON`);
}
/** @internal */
export const Error3$inboundSchema = z
    .object({
    code: GetOrderErrorDomainsRegistrarCode$inboundSchema,
    details: z.lazy(() => ErrorDetails$inboundSchema),
});
/** @internal */
export const Error3$outboundSchema = z.object({
    code: GetOrderErrorDomainsRegistrarCode$outboundSchema,
    details: z.lazy(() => ErrorDetails$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Error3$;
(function (Error3$) {
    /** @deprecated use `Error3$inboundSchema` instead. */
    Error3$.inboundSchema = Error3$inboundSchema;
    /** @deprecated use `Error3$outboundSchema` instead. */
    Error3$.outboundSchema = Error3$outboundSchema;
})(Error3$ || (Error3$ = {}));
export function error3ToJSON(error3) {
    return JSON.stringify(Error3$outboundSchema.parse(error3));
}
export function error3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Error3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Error3' from JSON`);
}
/** @internal */
export const GetOrderErrorCode$inboundSchema = z.nativeEnum(GetOrderErrorCode);
/** @internal */
export const GetOrderErrorCode$outboundSchema = GetOrderErrorCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderErrorCode$;
(function (GetOrderErrorCode$) {
    /** @deprecated use `GetOrderErrorCode$inboundSchema` instead. */
    GetOrderErrorCode$.inboundSchema = GetOrderErrorCode$inboundSchema;
    /** @deprecated use `GetOrderErrorCode$outboundSchema` instead. */
    GetOrderErrorCode$.outboundSchema = GetOrderErrorCode$outboundSchema;
})(GetOrderErrorCode$ || (GetOrderErrorCode$ = {}));
/** @internal */
export const Tlds$inboundSchema = z
    .object({
    tldName: z.string(),
    endsAt: z.string(),
});
/** @internal */
export const Tlds$outboundSchema = z.object({
    tldName: z.string(),
    endsAt: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Tlds$;
(function (Tlds$) {
    /** @deprecated use `Tlds$inboundSchema` instead. */
    Tlds$.inboundSchema = Tlds$inboundSchema;
    /** @deprecated use `Tlds$outboundSchema` instead. */
    Tlds$.outboundSchema = Tlds$outboundSchema;
})(Tlds$ || (Tlds$ = {}));
export function tldsToJSON(tlds) {
    return JSON.stringify(Tlds$outboundSchema.parse(tlds));
}
export function tldsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Tlds$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Tlds' from JSON`);
}
/** @internal */
export const GetOrderErrorDetails$inboundSchema = z.object({
    tlds: z.array(z.lazy(() => Tlds$inboundSchema)),
});
/** @internal */
export const GetOrderErrorDetails$outboundSchema = z.object({
    tlds: z.array(z.lazy(() => Tlds$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderErrorDetails$;
(function (GetOrderErrorDetails$) {
    /** @deprecated use `GetOrderErrorDetails$inboundSchema` instead. */
    GetOrderErrorDetails$.inboundSchema = GetOrderErrorDetails$inboundSchema;
    /** @deprecated use `GetOrderErrorDetails$outboundSchema` instead. */
    GetOrderErrorDetails$.outboundSchema = GetOrderErrorDetails$outboundSchema;
})(GetOrderErrorDetails$ || (GetOrderErrorDetails$ = {}));
export function getOrderErrorDetailsToJSON(getOrderErrorDetails) {
    return JSON.stringify(GetOrderErrorDetails$outboundSchema.parse(getOrderErrorDetails));
}
export function getOrderErrorDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderErrorDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderErrorDetails' from JSON`);
}
/** @internal */
export const Error2$inboundSchema = z
    .object({
    code: GetOrderErrorCode$inboundSchema,
    details: z.lazy(() => GetOrderErrorDetails$inboundSchema),
});
/** @internal */
export const Error2$outboundSchema = z.object({
    code: GetOrderErrorCode$outboundSchema,
    details: z.lazy(() => GetOrderErrorDetails$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Error2$;
(function (Error2$) {
    /** @deprecated use `Error2$inboundSchema` instead. */
    Error2$.inboundSchema = Error2$inboundSchema;
    /** @deprecated use `Error2$outboundSchema` instead. */
    Error2$.outboundSchema = Error2$outboundSchema;
})(Error2$ || (Error2$ = {}));
export function error2ToJSON(error2) {
    return JSON.stringify(Error2$outboundSchema.parse(error2));
}
export function error2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Error2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Error2' from JSON`);
}
/** @internal */
export const ErrorCode$inboundSchema = z
    .nativeEnum(ErrorCode);
/** @internal */
export const ErrorCode$outboundSchema = ErrorCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ErrorCode$;
(function (ErrorCode$) {
    /** @deprecated use `ErrorCode$inboundSchema` instead. */
    ErrorCode$.inboundSchema = ErrorCode$inboundSchema;
    /** @deprecated use `ErrorCode$outboundSchema` instead. */
    ErrorCode$.outboundSchema = ErrorCode$outboundSchema;
})(ErrorCode$ || (ErrorCode$ = {}));
/** @internal */
export const Error1$inboundSchema = z
    .object({
    code: ErrorCode$inboundSchema,
});
/** @internal */
export const Error1$outboundSchema = z.object({
    code: ErrorCode$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Error1$;
(function (Error1$) {
    /** @deprecated use `Error1$inboundSchema` instead. */
    Error1$.inboundSchema = Error1$inboundSchema;
    /** @deprecated use `Error1$outboundSchema` instead. */
    Error1$.outboundSchema = Error1$outboundSchema;
})(Error1$ || (Error1$ = {}));
export function error1ToJSON(error1) {
    return JSON.stringify(Error1$outboundSchema.parse(error1));
}
export function error1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Error1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Error1' from JSON`);
}
/** @internal */
export const ErrorT$inboundSchema = z
    .union([
    z.lazy(() => Error2$inboundSchema),
    z.lazy(() => Error3$inboundSchema),
    z.lazy(() => Error1$inboundSchema),
    z.lazy(() => Error4$inboundSchema),
]);
/** @internal */
export const ErrorT$outboundSchema = z.union([
    z.lazy(() => Error2$outboundSchema),
    z.lazy(() => Error3$outboundSchema),
    z.lazy(() => Error1$outboundSchema),
    z.lazy(() => Error4$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ErrorT$;
(function (ErrorT$) {
    /** @deprecated use `ErrorT$inboundSchema` instead. */
    ErrorT$.inboundSchema = ErrorT$inboundSchema;
    /** @deprecated use `ErrorT$outboundSchema` instead. */
    ErrorT$.outboundSchema = ErrorT$outboundSchema;
})(ErrorT$ || (ErrorT$ = {}));
export function errorToJSON(errorT) {
    return JSON.stringify(ErrorT$outboundSchema.parse(errorT));
}
export function errorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ErrorT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ErrorT' from JSON`);
}
/** @internal */
export const GetOrderResponseBody$inboundSchema = z.object({
    orderId: z.string(),
    domains: z.array(z.union([
        z.lazy(() => Domains1$inboundSchema),
        z.lazy(() => Domains3$inboundSchema),
        z.lazy(() => Domains2$inboundSchema),
    ])),
    status: GetOrderStatus$inboundSchema,
    error: z.union([
        z.lazy(() => Error2$inboundSchema),
        z.lazy(() => Error3$inboundSchema),
        z.lazy(() => Error1$inboundSchema),
        z.lazy(() => Error4$inboundSchema),
    ]).optional(),
});
/** @internal */
export const GetOrderResponseBody$outboundSchema = z.object({
    orderId: z.string(),
    domains: z.array(z.union([
        z.lazy(() => Domains1$outboundSchema),
        z.lazy(() => Domains3$outboundSchema),
        z.lazy(() => Domains2$outboundSchema),
    ])),
    status: GetOrderStatus$outboundSchema,
    error: z.union([
        z.lazy(() => Error2$outboundSchema),
        z.lazy(() => Error3$outboundSchema),
        z.lazy(() => Error1$outboundSchema),
        z.lazy(() => Error4$outboundSchema),
    ]).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetOrderResponseBody$;
(function (GetOrderResponseBody$) {
    /** @deprecated use `GetOrderResponseBody$inboundSchema` instead. */
    GetOrderResponseBody$.inboundSchema = GetOrderResponseBody$inboundSchema;
    /** @deprecated use `GetOrderResponseBody$outboundSchema` instead. */
    GetOrderResponseBody$.outboundSchema = GetOrderResponseBody$outboundSchema;
})(GetOrderResponseBody$ || (GetOrderResponseBody$ = {}));
export function getOrderResponseBodyToJSON(getOrderResponseBody) {
    return JSON.stringify(GetOrderResponseBody$outboundSchema.parse(getOrderResponseBody));
}
export function getOrderResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderResponseBody' from JSON`);
}
