// @vercel/sdk/esm/models/getsupportedtldsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetSupportedTldsRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
});
/** @internal */
export const GetSupportedTldsRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetSupportedTldsRequest$;
(function (GetSupportedTldsRequest$) {
    /** @deprecated use `GetSupportedTldsRequest$inboundSchema` instead. */
    GetSupportedTldsRequest$.inboundSchema = GetSupportedTldsRequest$inboundSchema;
    /** @deprecated use `GetSupportedTldsRequest$outboundSchema` instead. */
    GetSupportedTldsRequest$.outboundSchema = GetSupportedTldsRequest$outboundSchema;
})(GetSupportedTldsRequest$ || (GetSupportedTldsRequest$ = {}));
export function getSupportedTldsRequestToJSON(getSupportedTldsRequest) {
    return JSON.stringify(GetSupportedTldsRequest$outboundSchema.parse(getSupportedTldsRequest));
}
export function getSupportedTldsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetSupportedTldsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSupportedTldsRequest' from JSON`);
}
