// @vercel/sdk/esm/models/gettldpriceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { TldNotSupported$inboundSchema, TldNotSupported$outboundSchema, } from "./tldnotsupported.js";
/** @internal */
export const GetTldPriceRequest$inboundSchema = z.object({
    tld: z.string(),
    years: z.string().optional(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetTldPriceRequest$outboundSchema = z.object({
    tld: z.string(),
    years: z.string().optional(),
    teamId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetTldPriceRequest$;
(function (GetTldPriceRequest$) {
    /** @deprecated use `GetTldPriceRequest$inboundSchema` instead. */
    GetTldPriceRequest$.inboundSchema = GetTldPriceRequest$inboundSchema;
    /** @deprecated use `GetTldPriceRequest$outboundSchema` instead. */
    GetTldPriceRequest$.outboundSchema = GetTldPriceRequest$outboundSchema;
})(GetTldPriceRequest$ || (GetTldPriceRequest$ = {}));
export function getTldPriceRequestToJSON(getTldPriceRequest) {
    return JSON.stringify(GetTldPriceRequest$outboundSchema.parse(getTldPriceRequest));
}
export function getTldPriceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTldPriceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTldPriceRequest' from JSON`);
}
/** @internal */
export const GetTldPriceDomainsRegistrarResponseBody$inboundSchema = z.union([TldNotSupported$inboundSchema, HttpApiDecodeError$inboundSchema]);
/** @internal */
export const GetTldPriceDomainsRegistrarResponseBody$outboundSchema = z.union([
    TldNotSupported$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetTldPriceDomainsRegistrarResponseBody$;
(function (GetTldPriceDomainsRegistrarResponseBody$) {
    /** @deprecated use `GetTldPriceDomainsRegistrarResponseBody$inboundSchema` instead. */
    GetTldPriceDomainsRegistrarResponseBody$.inboundSchema = GetTldPriceDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `GetTldPriceDomainsRegistrarResponseBody$outboundSchema` instead. */
    GetTldPriceDomainsRegistrarResponseBody$.outboundSchema = GetTldPriceDomainsRegistrarResponseBody$outboundSchema;
})(GetTldPriceDomainsRegistrarResponseBody$ || (GetTldPriceDomainsRegistrarResponseBody$ = {}));
export function getTldPriceDomainsRegistrarResponseBodyToJSON(getTldPriceDomainsRegistrarResponseBody) {
    return JSON.stringify(GetTldPriceDomainsRegistrarResponseBody$outboundSchema.parse(getTldPriceDomainsRegistrarResponseBody));
}
export function getTldPriceDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTldPriceDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTldPriceDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetTldPriceResponseBody$inboundSchema = z.object({
    years: z.number(),
    purchasePrice: z.nullable(z.number()),
    renewalPrice: z.nullable(z.number()),
    transferPrice: z.nullable(z.number()),
});
/** @internal */
export const GetTldPriceResponseBody$outboundSchema = z.object({
    years: z.number(),
    purchasePrice: z.nullable(z.number()),
    renewalPrice: z.nullable(z.number()),
    transferPrice: z.nullable(z.number()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetTldPriceResponseBody$;
(function (GetTldPriceResponseBody$) {
    /** @deprecated use `GetTldPriceResponseBody$inboundSchema` instead. */
    GetTldPriceResponseBody$.inboundSchema = GetTldPriceResponseBody$inboundSchema;
    /** @deprecated use `GetTldPriceResponseBody$outboundSchema` instead. */
    GetTldPriceResponseBody$.outboundSchema = GetTldPriceResponseBody$outboundSchema;
})(GetTldPriceResponseBody$ || (GetTldPriceResponseBody$ = {}));
export function getTldPriceResponseBodyToJSON(getTldPriceResponseBody) {
    return JSON.stringify(GetTldPriceResponseBody$outboundSchema.parse(getTldPriceResponseBody));
}
export function getTldPriceResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTldPriceResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTldPriceResponseBody' from JSON`);
}
