// @vercel/sdk/esm/models/internalservererror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const InternalServerErrorCode = {
    InternalServerError: "internal_server_error",
};
export class InternalServerError extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "InternalServerError";
    }
}
/** @internal */
export const InternalServerErrorCode$inboundSchema = z.nativeEnum(InternalServerErrorCode);
/** @internal */
export const InternalServerErrorCode$outboundSchema = InternalServerErrorCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InternalServerErrorCode$;
(function (InternalServerErrorCode$) {
    /** @deprecated use `InternalServerErrorCode$inboundSchema` instead. */
    InternalServerErrorCode$.inboundSchema = InternalServerErrorCode$inboundSchema;
    /** @deprecated use `InternalServerErrorCode$outboundSchema` instead. */
    InternalServerErrorCode$.outboundSchema = InternalServerErrorCode$outboundSchema;
})(InternalServerErrorCode$ || (InternalServerErrorCode$ = {}));
/** @internal */
export const InternalServerError$inboundSchema = z.object({
    status: z.number(),
    code: InternalServerErrorCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new InternalServerError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const InternalServerError$outboundSchema = z.instanceof(InternalServerError)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: InternalServerErrorCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InternalServerError$;
(function (InternalServerError$) {
    /** @deprecated use `InternalServerError$inboundSchema` instead. */
    InternalServerError$.inboundSchema = InternalServerError$inboundSchema;
    /** @deprecated use `InternalServerError$outboundSchema` instead. */
    InternalServerError$.outboundSchema = InternalServerError$outboundSchema;
})(InternalServerError$ || (InternalServerError$ = {}));
