// @vercel/sdk/esm/models/invalidatebytagsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const InvalidateByTagsTarget = {
    Production: "production",
    Preview: "preview",
};
/** @internal */
export const Tags$inboundSchema = z
    .union([z.array(z.string()), z.string()]);
/** @internal */
export const Tags$outboundSchema = z.union([z.array(z.string()), z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Tags$;
(function (Tags$) {
    /** @deprecated use `Tags$inboundSchema` instead. */
    Tags$.inboundSchema = Tags$inboundSchema;
    /** @deprecated use `Tags$outboundSchema` instead. */
    Tags$.outboundSchema = Tags$outboundSchema;
})(Tags$ || (Tags$ = {}));
export function tagsToJSON(tags) {
    return JSON.stringify(Tags$outboundSchema.parse(tags));
}
export function tagsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Tags$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Tags' from JSON`);
}
/** @internal */
export const InvalidateByTagsTarget$inboundSchema = z.nativeEnum(InvalidateByTagsTarget);
/** @internal */
export const InvalidateByTagsTarget$outboundSchema = InvalidateByTagsTarget$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InvalidateByTagsTarget$;
(function (InvalidateByTagsTarget$) {
    /** @deprecated use `InvalidateByTagsTarget$inboundSchema` instead. */
    InvalidateByTagsTarget$.inboundSchema = InvalidateByTagsTarget$inboundSchema;
    /** @deprecated use `InvalidateByTagsTarget$outboundSchema` instead. */
    InvalidateByTagsTarget$.outboundSchema = InvalidateByTagsTarget$outboundSchema;
})(InvalidateByTagsTarget$ || (InvalidateByTagsTarget$ = {}));
/** @internal */
export const InvalidateByTagsRequestBody$inboundSchema = z.object({
    tags: z.union([z.array(z.string()), z.string()]),
    target: InvalidateByTagsTarget$inboundSchema.optional(),
});
/** @internal */
export const InvalidateByTagsRequestBody$outboundSchema = z.object({
    tags: z.union([z.array(z.string()), z.string()]),
    target: InvalidateByTagsTarget$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InvalidateByTagsRequestBody$;
(function (InvalidateByTagsRequestBody$) {
    /** @deprecated use `InvalidateByTagsRequestBody$inboundSchema` instead. */
    InvalidateByTagsRequestBody$.inboundSchema = InvalidateByTagsRequestBody$inboundSchema;
    /** @deprecated use `InvalidateByTagsRequestBody$outboundSchema` instead. */
    InvalidateByTagsRequestBody$.outboundSchema = InvalidateByTagsRequestBody$outboundSchema;
})(InvalidateByTagsRequestBody$ || (InvalidateByTagsRequestBody$ = {}));
export function invalidateByTagsRequestBodyToJSON(invalidateByTagsRequestBody) {
    return JSON.stringify(InvalidateByTagsRequestBody$outboundSchema.parse(invalidateByTagsRequestBody));
}
export function invalidateByTagsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InvalidateByTagsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InvalidateByTagsRequestBody' from JSON`);
}
/** @internal */
export const InvalidateByTagsRequest$inboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => InvalidateByTagsRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const InvalidateByTagsRequest$outboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => InvalidateByTagsRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InvalidateByTagsRequest$;
(function (InvalidateByTagsRequest$) {
    /** @deprecated use `InvalidateByTagsRequest$inboundSchema` instead. */
    InvalidateByTagsRequest$.inboundSchema = InvalidateByTagsRequest$inboundSchema;
    /** @deprecated use `InvalidateByTagsRequest$outboundSchema` instead. */
    InvalidateByTagsRequest$.outboundSchema = InvalidateByTagsRequest$outboundSchema;
})(InvalidateByTagsRequest$ || (InvalidateByTagsRequest$ = {}));
export function invalidateByTagsRequestToJSON(invalidateByTagsRequest) {
    return JSON.stringify(InvalidateByTagsRequest$outboundSchema.parse(invalidateByTagsRequest));
}
export function invalidateByTagsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InvalidateByTagsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InvalidateByTagsRequest' from JSON`);
}
