// @vercel/sdk/esm/models/invalidadditionalcontactinfo.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const InvalidAdditionalContactInfoCode = {
    InvalidAdditionalContactInfo: "invalid_additional_contact_info",
};
/**
 * Additional contact information provided for the TLD is invalid.
 */
export class InvalidAdditionalContactInfo extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "InvalidAdditionalContactInfo";
    }
}
/** @internal */
export const InvalidAdditionalContactInfoCode$inboundSchema = z.nativeEnum(InvalidAdditionalContactInfoCode);
/** @internal */
export const InvalidAdditionalContactInfoCode$outboundSchema = InvalidAdditionalContactInfoCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InvalidAdditionalContactInfoCode$;
(function (InvalidAdditionalContactInfoCode$) {
    /** @deprecated use `InvalidAdditionalContactInfoCode$inboundSchema` instead. */
    InvalidAdditionalContactInfoCode$.inboundSchema = InvalidAdditionalContactInfoCode$inboundSchema;
    /** @deprecated use `InvalidAdditionalContactInfoCode$outboundSchema` instead. */
    InvalidAdditionalContactInfoCode$.outboundSchema = InvalidAdditionalContactInfoCode$outboundSchema;
})(InvalidAdditionalContactInfoCode$ || (InvalidAdditionalContactInfoCode$ = {}));
/** @internal */
export const InvalidAdditionalContactInfo$inboundSchema = z.object({
    status: z.number(),
    code: InvalidAdditionalContactInfoCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new InvalidAdditionalContactInfo(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const InvalidAdditionalContactInfo$outboundSchema = z.instanceof(InvalidAdditionalContactInfo)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: InvalidAdditionalContactInfoCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InvalidAdditionalContactInfo$;
(function (InvalidAdditionalContactInfo$) {
    /** @deprecated use `InvalidAdditionalContactInfo$inboundSchema` instead. */
    InvalidAdditionalContactInfo$.inboundSchema = InvalidAdditionalContactInfo$inboundSchema;
    /** @deprecated use `InvalidAdditionalContactInfo$outboundSchema` instead. */
    InvalidAdditionalContactInfo$.outboundSchema = InvalidAdditionalContactInfo$outboundSchema;
})(InvalidAdditionalContactInfo$ || (InvalidAdditionalContactInfo$ = {}));
