// @vercel/sdk/esm/models/issue.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { PropertyKey$inboundSchema, PropertyKey$outboundSchema, } from "./propertykey.js";
/** @internal */
export const Issue$inboundSchema = z
    .object({
    path: z.array(PropertyKey$inboundSchema),
    message: z.string(),
});
/** @internal */
export const Issue$outboundSchema = z.object({
    path: z.array(PropertyKey$outboundSchema),
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Issue$;
(function (Issue$) {
    /** @deprecated use `Issue$inboundSchema` instead. */
    Issue$.inboundSchema = Issue$inboundSchema;
    /** @deprecated use `Issue$outboundSchema` instead. */
    Issue$.outboundSchema = Issue$outboundSchema;
})(Issue$ || (Issue$ = {}));
export function issueToJSON(issue) {
    return JSON.stringify(Issue$outboundSchema.parse(issue));
}
export function issueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Issue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Issue' from JSON`);
}
