// @vercel/sdk/esm/models/notauthorizedforscope.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const NotAuthorizedForScopeCode = {
    NotAuthorizedForScope: "not_authorized_for_scope",
};
export class NotAuthorizedForScope extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "NotAuthorizedForScope";
    }
}
/** @internal */
export const NotAuthorizedForScopeCode$inboundSchema = z.nativeEnum(NotAuthorizedForScopeCode);
/** @internal */
export const NotAuthorizedForScopeCode$outboundSchema = NotAuthorizedForScopeCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotAuthorizedForScopeCode$;
(function (NotAuthorizedForScopeCode$) {
    /** @deprecated use `NotAuthorizedForScopeCode$inboundSchema` instead. */
    NotAuthorizedForScopeCode$.inboundSchema = NotAuthorizedForScopeCode$inboundSchema;
    /** @deprecated use `NotAuthorizedForScopeCode$outboundSchema` instead. */
    NotAuthorizedForScopeCode$.outboundSchema = NotAuthorizedForScopeCode$outboundSchema;
})(NotAuthorizedForScopeCode$ || (NotAuthorizedForScopeCode$ = {}));
/** @internal */
export const NotAuthorizedForScope$inboundSchema = z.object({
    status: z.number(),
    code: NotAuthorizedForScopeCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new NotAuthorizedForScope(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const NotAuthorizedForScope$outboundSchema = z.instanceof(NotAuthorizedForScope)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: NotAuthorizedForScopeCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotAuthorizedForScope$;
(function (NotAuthorizedForScope$) {
    /** @deprecated use `NotAuthorizedForScope$inboundSchema` instead. */
    NotAuthorizedForScope$.inboundSchema = NotAuthorizedForScope$inboundSchema;
    /** @deprecated use `NotAuthorizedForScope$outboundSchema` instead. */
    NotAuthorizedForScope$.outboundSchema = NotAuthorizedForScope$outboundSchema;
})(NotAuthorizedForScope$ || (NotAuthorizedForScope$ = {}));
