// @vercel/sdk/esm/models/notfound.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const NotFoundCode = {
    NotFound: "not_found",
};
export class NotFound extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "NotFound";
    }
}
/** @internal */
export const NotFoundCode$inboundSchema = z.nativeEnum(NotFoundCode);
/** @internal */
export const NotFoundCode$outboundSchema = NotFoundCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotFoundCode$;
(function (NotFoundCode$) {
    /** @deprecated use `NotFoundCode$inboundSchema` instead. */
    NotFoundCode$.inboundSchema = NotFoundCode$inboundSchema;
    /** @deprecated use `NotFoundCode$outboundSchema` instead. */
    NotFoundCode$.outboundSchema = NotFoundCode$outboundSchema;
})(NotFoundCode$ || (NotFoundCode$ = {}));
/** @internal */
export const NotFound$inboundSchema = z.object({
    status: z.number(),
    code: NotFoundCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new NotFound(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const NotFound$outboundSchema = z.instanceof(NotFound)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: NotFoundCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var NotFound$;
(function (NotFound$) {
    /** @deprecated use `NotFound$inboundSchema` instead. */
    NotFound$.inboundSchema = NotFound$inboundSchema;
    /** @deprecated use `NotFound$outboundSchema` instead. */
    NotFound$.outboundSchema = NotFound$outboundSchema;
})(NotFound$ || (NotFound$ = {}));
