// @vercel/sdk/esm/models/ordertooexpensive.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const OrderTooExpensiveCode = {
    OrderTooExpensive: "order_too_expensive",
};
/**
 * The total price of the order is too high.
 */
export class OrderTooExpensive extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "OrderTooExpensive";
    }
}
/** @internal */
export const OrderTooExpensiveCode$inboundSchema = z.nativeEnum(OrderTooExpensiveCode);
/** @internal */
export const OrderTooExpensiveCode$outboundSchema = OrderTooExpensiveCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrderTooExpensiveCode$;
(function (OrderTooExpensiveCode$) {
    /** @deprecated use `OrderTooExpensiveCode$inboundSchema` instead. */
    OrderTooExpensiveCode$.inboundSchema = OrderTooExpensiveCode$inboundSchema;
    /** @deprecated use `OrderTooExpensiveCode$outboundSchema` instead. */
    OrderTooExpensiveCode$.outboundSchema = OrderTooExpensiveCode$outboundSchema;
})(OrderTooExpensiveCode$ || (OrderTooExpensiveCode$ = {}));
/** @internal */
export const OrderTooExpensive$inboundSchema = z.object({
    status: z.number(),
    code: OrderTooExpensiveCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new OrderTooExpensive(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const OrderTooExpensive$outboundSchema = z.instanceof(OrderTooExpensive)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: OrderTooExpensiveCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrderTooExpensive$;
(function (OrderTooExpensive$) {
    /** @deprecated use `OrderTooExpensive$inboundSchema` instead. */
    OrderTooExpensive$.inboundSchema = OrderTooExpensive$inboundSchema;
    /** @deprecated use `OrderTooExpensive$outboundSchema` instead. */
    OrderTooExpensive$.outboundSchema = OrderTooExpensive$outboundSchema;
})(OrderTooExpensive$ || (OrderTooExpensive$ = {}));
