// @vercel/sdk/esm/models/propertykey.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const Tag = {
    Symbol: "symbol",
};
/** @internal */
export const Tag$inboundSchema = z.nativeEnum(Tag);
/** @internal */
export const Tag$outboundSchema = Tag$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Tag$;
(function (Tag$) {
    /** @deprecated use `Tag$inboundSchema` instead. */
    Tag$.inboundSchema = Tag$inboundSchema;
    /** @deprecated use `Tag$outboundSchema` instead. */
    Tag$.outboundSchema = Tag$outboundSchema;
})(Tag$ || (Tag$ = {}));
/** @internal */
export const PropertyKey3$inboundSchema = z.object({
    _tag: Tag$inboundSchema,
    key: z.string(),
}).transform((v) => {
    return remap$(v, {
        "_tag": "tag",
    });
});
/** @internal */
export const PropertyKey3$outboundSchema = z.object({
    tag: Tag$outboundSchema,
    key: z.string(),
}).transform((v) => {
    return remap$(v, {
        tag: "_tag",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PropertyKey3$;
(function (PropertyKey3$) {
    /** @deprecated use `PropertyKey3$inboundSchema` instead. */
    PropertyKey3$.inboundSchema = PropertyKey3$inboundSchema;
    /** @deprecated use `PropertyKey3$outboundSchema` instead. */
    PropertyKey3$.outboundSchema = PropertyKey3$outboundSchema;
})(PropertyKey3$ || (PropertyKey3$ = {}));
export function propertyKey3ToJSON(propertyKey3) {
    return JSON.stringify(PropertyKey3$outboundSchema.parse(propertyKey3));
}
export function propertyKey3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => PropertyKey3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PropertyKey3' from JSON`);
}
/** @internal */
export const PropertyKey$inboundSchema = z.union([z.lazy(() => PropertyKey3$inboundSchema), z.string(), z.number()]);
/** @internal */
export const PropertyKey$outboundSchema = z.union([
    z.lazy(() => PropertyKey3$outboundSchema),
    z.string(),
    z.number(),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PropertyKey$;
(function (PropertyKey$) {
    /** @deprecated use `PropertyKey$inboundSchema` instead. */
    PropertyKey$.inboundSchema = PropertyKey$inboundSchema;
    /** @deprecated use `PropertyKey$outboundSchema` instead. */
    PropertyKey$.outboundSchema = PropertyKey$outboundSchema;
})(PropertyKey$ || (PropertyKey$ = {}));
export function propertyKeyToJSON(propertyKey) {
    return JSON.stringify(PropertyKey$outboundSchema.parse(propertyKey));
}
export function propertyKeyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PropertyKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PropertyKey' from JSON`);
}
