// @vercel/sdk/esm/models/renewdomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { BadRequest$inboundSchema, BadRequest$outboundSchema, } from "./badrequest.js";
import { DomainNotAvailable$inboundSchema, DomainNotAvailable$outboundSchema, } from "./domainnotavailable.js";
import { DomainNotRegistered$inboundSchema, DomainNotRegistered$outboundSchema, } from "./domainnotregistered.js";
import { ExpectedPriceMismatch$inboundSchema, ExpectedPriceMismatch$outboundSchema, } from "./expectedpricemismatch.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
import { TldNotSupported$inboundSchema, TldNotSupported$outboundSchema, } from "./tldnotsupported.js";
export const RenewDomainMethod = {
    Get: "GET",
    Post: "POST",
    Put: "PUT",
    Delete: "DELETE",
    Patch: "PATCH",
};
/** @internal */
export const RenewDomainContactInformation$inboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    email: z.string(),
    phone: z.string(),
    address1: z.string(),
    address2: z.string().optional(),
    city: z.string(),
    state: z.string(),
    zip: z.string(),
    country: z.string(),
    companyName: z.string().optional(),
    fax: z.string().optional(),
});
/** @internal */
export const RenewDomainContactInformation$outboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    email: z.string(),
    phone: z.string(),
    address1: z.string(),
    address2: z.string().optional(),
    city: z.string(),
    state: z.string(),
    zip: z.string(),
    country: z.string(),
    companyName: z.string().optional(),
    fax: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainContactInformation$;
(function (RenewDomainContactInformation$) {
    /** @deprecated use `RenewDomainContactInformation$inboundSchema` instead. */
    RenewDomainContactInformation$.inboundSchema = RenewDomainContactInformation$inboundSchema;
    /** @deprecated use `RenewDomainContactInformation$outboundSchema` instead. */
    RenewDomainContactInformation$.outboundSchema = RenewDomainContactInformation$outboundSchema;
})(RenewDomainContactInformation$ || (RenewDomainContactInformation$ = {}));
export function renewDomainContactInformationToJSON(renewDomainContactInformation) {
    return JSON.stringify(RenewDomainContactInformation$outboundSchema.parse(renewDomainContactInformation));
}
export function renewDomainContactInformationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainContactInformation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainContactInformation' from JSON`);
}
/** @internal */
export const RenewDomainRequestBody$inboundSchema = z.object({
    years: z.number(),
    expectedPrice: z.number(),
    contactInformation: z.lazy(() => RenewDomainContactInformation$inboundSchema)
        .optional(),
});
/** @internal */
export const RenewDomainRequestBody$outboundSchema = z.object({
    years: z.number(),
    expectedPrice: z.number(),
    contactInformation: z.lazy(() => RenewDomainContactInformation$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainRequestBody$;
(function (RenewDomainRequestBody$) {
    /** @deprecated use `RenewDomainRequestBody$inboundSchema` instead. */
    RenewDomainRequestBody$.inboundSchema = RenewDomainRequestBody$inboundSchema;
    /** @deprecated use `RenewDomainRequestBody$outboundSchema` instead. */
    RenewDomainRequestBody$.outboundSchema = RenewDomainRequestBody$outboundSchema;
})(RenewDomainRequestBody$ || (RenewDomainRequestBody$ = {}));
export function renewDomainRequestBodyToJSON(renewDomainRequestBody) {
    return JSON.stringify(RenewDomainRequestBody$outboundSchema.parse(renewDomainRequestBody));
}
export function renewDomainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainRequestBody' from JSON`);
}
/** @internal */
export const RenewDomainRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    RequestBody: z.lazy(() => RenewDomainRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const RenewDomainRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    requestBody: z.lazy(() => RenewDomainRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainRequest$;
(function (RenewDomainRequest$) {
    /** @deprecated use `RenewDomainRequest$inboundSchema` instead. */
    RenewDomainRequest$.inboundSchema = RenewDomainRequest$inboundSchema;
    /** @deprecated use `RenewDomainRequest$outboundSchema` instead. */
    RenewDomainRequest$.outboundSchema = RenewDomainRequest$outboundSchema;
})(RenewDomainRequest$ || (RenewDomainRequest$ = {}));
export function renewDomainRequestToJSON(renewDomainRequest) {
    return JSON.stringify(RenewDomainRequest$outboundSchema.parse(renewDomainRequest));
}
export function renewDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainRequest' from JSON`);
}
/** @internal */
export const RenewDomainDomainsRegistrarResponseResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const RenewDomainDomainsRegistrarResponseResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainDomainsRegistrarResponseResponseBody$;
(function (RenewDomainDomainsRegistrarResponseResponseBody$) {
    /** @deprecated use `RenewDomainDomainsRegistrarResponseResponseBody$inboundSchema` instead. */
    RenewDomainDomainsRegistrarResponseResponseBody$.inboundSchema = RenewDomainDomainsRegistrarResponseResponseBody$inboundSchema;
    /** @deprecated use `RenewDomainDomainsRegistrarResponseResponseBody$outboundSchema` instead. */
    RenewDomainDomainsRegistrarResponseResponseBody$.outboundSchema = RenewDomainDomainsRegistrarResponseResponseBody$outboundSchema;
})(RenewDomainDomainsRegistrarResponseResponseBody$ || (RenewDomainDomainsRegistrarResponseResponseBody$ = {}));
export function renewDomainDomainsRegistrarResponseResponseBodyToJSON(renewDomainDomainsRegistrarResponseResponseBody) {
    return JSON.stringify(RenewDomainDomainsRegistrarResponseResponseBody$outboundSchema.parse(renewDomainDomainsRegistrarResponseResponseBody));
}
export function renewDomainDomainsRegistrarResponseResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainDomainsRegistrarResponseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainDomainsRegistrarResponseResponseBody' from JSON`);
}
/** @internal */
export const RenewDomainDomainsRegistrarResponseBody$inboundSchema = z.union([
    BadRequest$inboundSchema,
    DomainNotRegistered$inboundSchema,
    ExpectedPriceMismatch$inboundSchema,
    DomainNotAvailable$inboundSchema,
    TldNotSupported$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const RenewDomainDomainsRegistrarResponseBody$outboundSchema = z.union([
    BadRequest$outboundSchema,
    DomainNotRegistered$outboundSchema,
    ExpectedPriceMismatch$outboundSchema,
    DomainNotAvailable$outboundSchema,
    TldNotSupported$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainDomainsRegistrarResponseBody$;
(function (RenewDomainDomainsRegistrarResponseBody$) {
    /** @deprecated use `RenewDomainDomainsRegistrarResponseBody$inboundSchema` instead. */
    RenewDomainDomainsRegistrarResponseBody$.inboundSchema = RenewDomainDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `RenewDomainDomainsRegistrarResponseBody$outboundSchema` instead. */
    RenewDomainDomainsRegistrarResponseBody$.outboundSchema = RenewDomainDomainsRegistrarResponseBody$outboundSchema;
})(RenewDomainDomainsRegistrarResponseBody$ || (RenewDomainDomainsRegistrarResponseBody$ = {}));
export function renewDomainDomainsRegistrarResponseBodyToJSON(renewDomainDomainsRegistrarResponseBody) {
    return JSON.stringify(RenewDomainDomainsRegistrarResponseBody$outboundSchema.parse(renewDomainDomainsRegistrarResponseBody));
}
export function renewDomainDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const RenewDomainMethod$inboundSchema = z.nativeEnum(RenewDomainMethod);
/** @internal */
export const RenewDomainMethod$outboundSchema = RenewDomainMethod$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainMethod$;
(function (RenewDomainMethod$) {
    /** @deprecated use `RenewDomainMethod$inboundSchema` instead. */
    RenewDomainMethod$.inboundSchema = RenewDomainMethod$inboundSchema;
    /** @deprecated use `RenewDomainMethod$outboundSchema` instead. */
    RenewDomainMethod$.outboundSchema = RenewDomainMethod$outboundSchema;
})(RenewDomainMethod$ || (RenewDomainMethod$ = {}));
/** @internal */
export const RenewDomainLinks$inboundSchema = z.object({
    href: z.string(),
    method: RenewDomainMethod$inboundSchema,
});
/** @internal */
export const RenewDomainLinks$outboundSchema = z.object({
    href: z.string(),
    method: RenewDomainMethod$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainLinks$;
(function (RenewDomainLinks$) {
    /** @deprecated use `RenewDomainLinks$inboundSchema` instead. */
    RenewDomainLinks$.inboundSchema = RenewDomainLinks$inboundSchema;
    /** @deprecated use `RenewDomainLinks$outboundSchema` instead. */
    RenewDomainLinks$.outboundSchema = RenewDomainLinks$outboundSchema;
})(RenewDomainLinks$ || (RenewDomainLinks$ = {}));
export function renewDomainLinksToJSON(renewDomainLinks) {
    return JSON.stringify(RenewDomainLinks$outboundSchema.parse(renewDomainLinks));
}
export function renewDomainLinksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainLinks$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainLinks' from JSON`);
}
/** @internal */
export const RenewDomainResponseBody$inboundSchema = z.object({
    orderId: z.string(),
    _links: z.record(z.lazy(() => RenewDomainLinks$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "_links": "links",
    });
});
/** @internal */
export const RenewDomainResponseBody$outboundSchema = z.object({
    orderId: z.string(),
    links: z.record(z.lazy(() => RenewDomainLinks$outboundSchema)),
}).transform((v) => {
    return remap$(v, {
        links: "_links",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RenewDomainResponseBody$;
(function (RenewDomainResponseBody$) {
    /** @deprecated use `RenewDomainResponseBody$inboundSchema` instead. */
    RenewDomainResponseBody$.inboundSchema = RenewDomainResponseBody$inboundSchema;
    /** @deprecated use `RenewDomainResponseBody$outboundSchema` instead. */
    RenewDomainResponseBody$.outboundSchema = RenewDomainResponseBody$outboundSchema;
})(RenewDomainResponseBody$ || (RenewDomainResponseBody$ = {}));
export function renewDomainResponseBodyToJSON(renewDomainResponseBody) {
    return JSON.stringify(RenewDomainResponseBody$outboundSchema.parse(renewDomainResponseBody));
}
export function renewDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewDomainResponseBody' from JSON`);
}
