// @vercel/sdk/esm/models/tldnotsupported.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { VercelError } from "./vercelerror.js";
export const TldNotSupportedCode = {
    TldNotSupported: "tld_not_supported",
};
/**
 * The TLD is not currently supported.
 */
export class TldNotSupported extends VercelError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.status = err.status;
        this.code = err.code;
        this.name = "TldNotSupported";
    }
}
/** @internal */
export const TldNotSupportedCode$inboundSchema = z.nativeEnum(TldNotSupportedCode);
/** @internal */
export const TldNotSupportedCode$outboundSchema = TldNotSupportedCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TldNotSupportedCode$;
(function (TldNotSupportedCode$) {
    /** @deprecated use `TldNotSupportedCode$inboundSchema` instead. */
    TldNotSupportedCode$.inboundSchema = TldNotSupportedCode$inboundSchema;
    /** @deprecated use `TldNotSupportedCode$outboundSchema` instead. */
    TldNotSupportedCode$.outboundSchema = TldNotSupportedCode$outboundSchema;
})(TldNotSupportedCode$ || (TldNotSupportedCode$ = {}));
/** @internal */
export const TldNotSupported$inboundSchema = z.object({
    status: z.number(),
    code: TldNotSupportedCode$inboundSchema,
    message: z.string(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new TldNotSupported(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const TldNotSupported$outboundSchema = z.instanceof(TldNotSupported)
    .transform(v => v.data$)
    .pipe(z.object({
    status: z.number(),
    code: TldNotSupportedCode$outboundSchema,
    message: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TldNotSupported$;
(function (TldNotSupported$) {
    /** @deprecated use `TldNotSupported$inboundSchema` instead. */
    TldNotSupported$.inboundSchema = TldNotSupported$inboundSchema;
    /** @deprecated use `TldNotSupported$outboundSchema` instead. */
    TldNotSupported$.outboundSchema = TldNotSupported$outboundSchema;
})(TldNotSupported$ || (TldNotSupported$ = {}));
