// @vercel/sdk/esm/models/transferindomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { BadRequest$inboundSchema, BadRequest$outboundSchema, } from "./badrequest.js";
import { DomainNotAvailable$inboundSchema, DomainNotAvailable$outboundSchema, } from "./domainnotavailable.js";
import { ExpectedPriceMismatch$inboundSchema, ExpectedPriceMismatch$outboundSchema, } from "./expectedpricemismatch.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
import { TldNotSupported$inboundSchema, TldNotSupported$outboundSchema, } from "./tldnotsupported.js";
export const TransferInDomainMethod = {
    Get: "GET",
    Post: "POST",
    Put: "PUT",
    Delete: "DELETE",
    Patch: "PATCH",
};
/** @internal */
export const TransferInDomainContactInformation$inboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    email: z.string(),
    phone: z.string(),
    address1: z.string(),
    address2: z.string().optional(),
    city: z.string(),
    state: z.string(),
    zip: z.string(),
    country: z.string(),
    companyName: z.string().optional(),
    fax: z.string().optional(),
});
/** @internal */
export const TransferInDomainContactInformation$outboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    email: z.string(),
    phone: z.string(),
    address1: z.string(),
    address2: z.string().optional(),
    city: z.string(),
    state: z.string(),
    zip: z.string(),
    country: z.string(),
    companyName: z.string().optional(),
    fax: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainContactInformation$;
(function (TransferInDomainContactInformation$) {
    /** @deprecated use `TransferInDomainContactInformation$inboundSchema` instead. */
    TransferInDomainContactInformation$.inboundSchema = TransferInDomainContactInformation$inboundSchema;
    /** @deprecated use `TransferInDomainContactInformation$outboundSchema` instead. */
    TransferInDomainContactInformation$.outboundSchema = TransferInDomainContactInformation$outboundSchema;
})(TransferInDomainContactInformation$ || (TransferInDomainContactInformation$ = {}));
export function transferInDomainContactInformationToJSON(transferInDomainContactInformation) {
    return JSON.stringify(TransferInDomainContactInformation$outboundSchema.parse(transferInDomainContactInformation));
}
export function transferInDomainContactInformationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainContactInformation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainContactInformation' from JSON`);
}
/** @internal */
export const TransferInDomainRequestBody$inboundSchema = z.object({
    authCode: z.string(),
    autoRenew: z.boolean(),
    years: z.number(),
    expectedPrice: z.number(),
    contactInformation: z.lazy(() => TransferInDomainContactInformation$inboundSchema),
});
/** @internal */
export const TransferInDomainRequestBody$outboundSchema = z.object({
    authCode: z.string(),
    autoRenew: z.boolean(),
    years: z.number(),
    expectedPrice: z.number(),
    contactInformation: z.lazy(() => TransferInDomainContactInformation$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainRequestBody$;
(function (TransferInDomainRequestBody$) {
    /** @deprecated use `TransferInDomainRequestBody$inboundSchema` instead. */
    TransferInDomainRequestBody$.inboundSchema = TransferInDomainRequestBody$inboundSchema;
    /** @deprecated use `TransferInDomainRequestBody$outboundSchema` instead. */
    TransferInDomainRequestBody$.outboundSchema = TransferInDomainRequestBody$outboundSchema;
})(TransferInDomainRequestBody$ || (TransferInDomainRequestBody$ = {}));
export function transferInDomainRequestBodyToJSON(transferInDomainRequestBody) {
    return JSON.stringify(TransferInDomainRequestBody$outboundSchema.parse(transferInDomainRequestBody));
}
export function transferInDomainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainRequestBody' from JSON`);
}
/** @internal */
export const TransferInDomainRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    RequestBody: z.lazy(() => TransferInDomainRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const TransferInDomainRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    requestBody: z.lazy(() => TransferInDomainRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainRequest$;
(function (TransferInDomainRequest$) {
    /** @deprecated use `TransferInDomainRequest$inboundSchema` instead. */
    TransferInDomainRequest$.inboundSchema = TransferInDomainRequest$inboundSchema;
    /** @deprecated use `TransferInDomainRequest$outboundSchema` instead. */
    TransferInDomainRequest$.outboundSchema = TransferInDomainRequest$outboundSchema;
})(TransferInDomainRequest$ || (TransferInDomainRequest$ = {}));
export function transferInDomainRequestToJSON(transferInDomainRequest) {
    return JSON.stringify(TransferInDomainRequest$outboundSchema.parse(transferInDomainRequest));
}
export function transferInDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainRequest' from JSON`);
}
/** @internal */
export const TransferInDomainDomainsRegistrarResponseResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const TransferInDomainDomainsRegistrarResponseResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainDomainsRegistrarResponseResponseBody$;
(function (TransferInDomainDomainsRegistrarResponseResponseBody$) {
    /** @deprecated use `TransferInDomainDomainsRegistrarResponseResponseBody$inboundSchema` instead. */
    TransferInDomainDomainsRegistrarResponseResponseBody$.inboundSchema = TransferInDomainDomainsRegistrarResponseResponseBody$inboundSchema;
    /** @deprecated use `TransferInDomainDomainsRegistrarResponseResponseBody$outboundSchema` instead. */
    TransferInDomainDomainsRegistrarResponseResponseBody$.outboundSchema = TransferInDomainDomainsRegistrarResponseResponseBody$outboundSchema;
})(TransferInDomainDomainsRegistrarResponseResponseBody$ || (TransferInDomainDomainsRegistrarResponseResponseBody$ = {}));
export function transferInDomainDomainsRegistrarResponseResponseBodyToJSON(transferInDomainDomainsRegistrarResponseResponseBody) {
    return JSON.stringify(TransferInDomainDomainsRegistrarResponseResponseBody$outboundSchema.parse(transferInDomainDomainsRegistrarResponseResponseBody));
}
export function transferInDomainDomainsRegistrarResponseResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainDomainsRegistrarResponseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainDomainsRegistrarResponseResponseBody' from JSON`);
}
/** @internal */
export const TransferInDomainDomainsRegistrarResponseBody$inboundSchema = z.union([
    BadRequest$inboundSchema,
    ExpectedPriceMismatch$inboundSchema,
    DomainNotAvailable$inboundSchema,
    TldNotSupported$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const TransferInDomainDomainsRegistrarResponseBody$outboundSchema = z.union([
    BadRequest$outboundSchema,
    ExpectedPriceMismatch$outboundSchema,
    DomainNotAvailable$outboundSchema,
    TldNotSupported$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainDomainsRegistrarResponseBody$;
(function (TransferInDomainDomainsRegistrarResponseBody$) {
    /** @deprecated use `TransferInDomainDomainsRegistrarResponseBody$inboundSchema` instead. */
    TransferInDomainDomainsRegistrarResponseBody$.inboundSchema = TransferInDomainDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `TransferInDomainDomainsRegistrarResponseBody$outboundSchema` instead. */
    TransferInDomainDomainsRegistrarResponseBody$.outboundSchema = TransferInDomainDomainsRegistrarResponseBody$outboundSchema;
})(TransferInDomainDomainsRegistrarResponseBody$ || (TransferInDomainDomainsRegistrarResponseBody$ = {}));
export function transferInDomainDomainsRegistrarResponseBodyToJSON(transferInDomainDomainsRegistrarResponseBody) {
    return JSON.stringify(TransferInDomainDomainsRegistrarResponseBody$outboundSchema.parse(transferInDomainDomainsRegistrarResponseBody));
}
export function transferInDomainDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const TransferInDomainMethod$inboundSchema = z.nativeEnum(TransferInDomainMethod);
/** @internal */
export const TransferInDomainMethod$outboundSchema = TransferInDomainMethod$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainMethod$;
(function (TransferInDomainMethod$) {
    /** @deprecated use `TransferInDomainMethod$inboundSchema` instead. */
    TransferInDomainMethod$.inboundSchema = TransferInDomainMethod$inboundSchema;
    /** @deprecated use `TransferInDomainMethod$outboundSchema` instead. */
    TransferInDomainMethod$.outboundSchema = TransferInDomainMethod$outboundSchema;
})(TransferInDomainMethod$ || (TransferInDomainMethod$ = {}));
/** @internal */
export const TransferInDomainLinks$inboundSchema = z.object({
    href: z.string(),
    method: TransferInDomainMethod$inboundSchema,
});
/** @internal */
export const TransferInDomainLinks$outboundSchema = z.object({
    href: z.string(),
    method: TransferInDomainMethod$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainLinks$;
(function (TransferInDomainLinks$) {
    /** @deprecated use `TransferInDomainLinks$inboundSchema` instead. */
    TransferInDomainLinks$.inboundSchema = TransferInDomainLinks$inboundSchema;
    /** @deprecated use `TransferInDomainLinks$outboundSchema` instead. */
    TransferInDomainLinks$.outboundSchema = TransferInDomainLinks$outboundSchema;
})(TransferInDomainLinks$ || (TransferInDomainLinks$ = {}));
export function transferInDomainLinksToJSON(transferInDomainLinks) {
    return JSON.stringify(TransferInDomainLinks$outboundSchema.parse(transferInDomainLinks));
}
export function transferInDomainLinksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainLinks$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainLinks' from JSON`);
}
/** @internal */
export const TransferInDomainResponseBody$inboundSchema = z.object({
    orderId: z.string(),
    _links: z.record(z.lazy(() => TransferInDomainLinks$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "_links": "links",
    });
});
/** @internal */
export const TransferInDomainResponseBody$outboundSchema = z.object({
    orderId: z.string(),
    links: z.record(z.lazy(() => TransferInDomainLinks$outboundSchema)),
}).transform((v) => {
    return remap$(v, {
        links: "_links",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TransferInDomainResponseBody$;
(function (TransferInDomainResponseBody$) {
    /** @deprecated use `TransferInDomainResponseBody$inboundSchema` instead. */
    TransferInDomainResponseBody$.inboundSchema = TransferInDomainResponseBody$inboundSchema;
    /** @deprecated use `TransferInDomainResponseBody$outboundSchema` instead. */
    TransferInDomainResponseBody$.outboundSchema = TransferInDomainResponseBody$outboundSchema;
})(TransferInDomainResponseBody$ || (TransferInDomainResponseBody$ = {}));
export function transferInDomainResponseBodyToJSON(transferInDomainResponseBody) {
    return JSON.stringify(TransferInDomainResponseBody$outboundSchema.parse(transferInDomainResponseBody));
}
export function transferInDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferInDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferInDomainResponseBody' from JSON`);
}
