// @vercel/sdk/esm/models/updateresourceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
export const UpdateResourceOwnership = {
    Owned: "owned",
    Linked: "linked",
    Sandbox: "sandbox",
};
export const UpdateResourceStatus = {
    Ready: "ready",
    Pending: "pending",
    Onboarding: "onboarding",
    Suspended: "suspended",
    Resumed: "resumed",
    Uninstalled: "uninstalled",
    Error: "error",
};
export const UpdateResourceType = {
    Prepayment: "prepayment",
    Subscription: "subscription",
};
export const UpdateResourceLevel = {
    Info: "info",
    Warn: "warn",
    Error: "error",
};
/** @internal */
export const UpdateResourceOwnership$inboundSchema = z.nativeEnum(UpdateResourceOwnership);
/** @internal */
export const UpdateResourceOwnership$outboundSchema = UpdateResourceOwnership$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceOwnership$;
(function (UpdateResourceOwnership$) {
    /** @deprecated use `UpdateResourceOwnership$inboundSchema` instead. */
    UpdateResourceOwnership$.inboundSchema = UpdateResourceOwnership$inboundSchema;
    /** @deprecated use `UpdateResourceOwnership$outboundSchema` instead. */
    UpdateResourceOwnership$.outboundSchema = UpdateResourceOwnership$outboundSchema;
})(UpdateResourceOwnership$ || (UpdateResourceOwnership$ = {}));
/** @internal */
export const UpdateResourceStatus$inboundSchema = z.nativeEnum(UpdateResourceStatus);
/** @internal */
export const UpdateResourceStatus$outboundSchema = UpdateResourceStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceStatus$;
(function (UpdateResourceStatus$) {
    /** @deprecated use `UpdateResourceStatus$inboundSchema` instead. */
    UpdateResourceStatus$.inboundSchema = UpdateResourceStatus$inboundSchema;
    /** @deprecated use `UpdateResourceStatus$outboundSchema` instead. */
    UpdateResourceStatus$.outboundSchema = UpdateResourceStatus$outboundSchema;
})(UpdateResourceStatus$ || (UpdateResourceStatus$ = {}));
/** @internal */
export const UpdateResourceType$inboundSchema = z.nativeEnum(UpdateResourceType);
/** @internal */
export const UpdateResourceType$outboundSchema = UpdateResourceType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceType$;
(function (UpdateResourceType$) {
    /** @deprecated use `UpdateResourceType$inboundSchema` instead. */
    UpdateResourceType$.inboundSchema = UpdateResourceType$inboundSchema;
    /** @deprecated use `UpdateResourceType$outboundSchema` instead. */
    UpdateResourceType$.outboundSchema = UpdateResourceType$outboundSchema;
})(UpdateResourceType$ || (UpdateResourceType$ = {}));
/** @internal */
export const UpdateResourceDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const UpdateResourceDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceDetails$;
(function (UpdateResourceDetails$) {
    /** @deprecated use `UpdateResourceDetails$inboundSchema` instead. */
    UpdateResourceDetails$.inboundSchema = UpdateResourceDetails$inboundSchema;
    /** @deprecated use `UpdateResourceDetails$outboundSchema` instead. */
    UpdateResourceDetails$.outboundSchema = UpdateResourceDetails$outboundSchema;
})(UpdateResourceDetails$ || (UpdateResourceDetails$ = {}));
export function updateResourceDetailsToJSON(updateResourceDetails) {
    return JSON.stringify(UpdateResourceDetails$outboundSchema.parse(updateResourceDetails));
}
export function updateResourceDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceDetails' from JSON`);
}
/** @internal */
export const UpdateResourceHighlightedDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const UpdateResourceHighlightedDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceHighlightedDetails$;
(function (UpdateResourceHighlightedDetails$) {
    /** @deprecated use `UpdateResourceHighlightedDetails$inboundSchema` instead. */
    UpdateResourceHighlightedDetails$.inboundSchema = UpdateResourceHighlightedDetails$inboundSchema;
    /** @deprecated use `UpdateResourceHighlightedDetails$outboundSchema` instead. */
    UpdateResourceHighlightedDetails$.outboundSchema = UpdateResourceHighlightedDetails$outboundSchema;
})(UpdateResourceHighlightedDetails$ || (UpdateResourceHighlightedDetails$ = {}));
export function updateResourceHighlightedDetailsToJSON(updateResourceHighlightedDetails) {
    return JSON.stringify(UpdateResourceHighlightedDetails$outboundSchema.parse(updateResourceHighlightedDetails));
}
export function updateResourceHighlightedDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceHighlightedDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceHighlightedDetails' from JSON`);
}
/** @internal */
export const UpdateResourceBillingPlan$inboundSchema = collectExtraKeys$(z.object({
    id: z.string(),
    type: UpdateResourceType$inboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => UpdateResourceDetails$inboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => UpdateResourceHighlightedDetails$inboundSchema)).optional(),
    effectiveDate: z.string().optional(),
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const UpdateResourceBillingPlan$outboundSchema = z.object({
    id: z.string(),
    type: UpdateResourceType$outboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => UpdateResourceDetails$outboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => UpdateResourceHighlightedDetails$outboundSchema)).optional(),
    effectiveDate: z.string().optional(),
    additionalProperties: z.record(z.any()),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceBillingPlan$;
(function (UpdateResourceBillingPlan$) {
    /** @deprecated use `UpdateResourceBillingPlan$inboundSchema` instead. */
    UpdateResourceBillingPlan$.inboundSchema = UpdateResourceBillingPlan$inboundSchema;
    /** @deprecated use `UpdateResourceBillingPlan$outboundSchema` instead. */
    UpdateResourceBillingPlan$.outboundSchema = UpdateResourceBillingPlan$outboundSchema;
})(UpdateResourceBillingPlan$ || (UpdateResourceBillingPlan$ = {}));
export function updateResourceBillingPlanToJSON(updateResourceBillingPlan) {
    return JSON.stringify(UpdateResourceBillingPlan$outboundSchema.parse(updateResourceBillingPlan));
}
export function updateResourceBillingPlanFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceBillingPlan$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceBillingPlan' from JSON`);
}
/** @internal */
export const UpdateResourceLevel$inboundSchema = z.nativeEnum(UpdateResourceLevel);
/** @internal */
export const UpdateResourceLevel$outboundSchema = UpdateResourceLevel$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceLevel$;
(function (UpdateResourceLevel$) {
    /** @deprecated use `UpdateResourceLevel$inboundSchema` instead. */
    UpdateResourceLevel$.inboundSchema = UpdateResourceLevel$inboundSchema;
    /** @deprecated use `UpdateResourceLevel$outboundSchema` instead. */
    UpdateResourceLevel$.outboundSchema = UpdateResourceLevel$outboundSchema;
})(UpdateResourceLevel$ || (UpdateResourceLevel$ = {}));
/** @internal */
export const UpdateResourceNotification$inboundSchema = z.object({
    level: UpdateResourceLevel$inboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/** @internal */
export const UpdateResourceNotification$outboundSchema = z.object({
    level: UpdateResourceLevel$outboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceNotification$;
(function (UpdateResourceNotification$) {
    /** @deprecated use `UpdateResourceNotification$inboundSchema` instead. */
    UpdateResourceNotification$.inboundSchema = UpdateResourceNotification$inboundSchema;
    /** @deprecated use `UpdateResourceNotification$outboundSchema` instead. */
    UpdateResourceNotification$.outboundSchema = UpdateResourceNotification$outboundSchema;
})(UpdateResourceNotification$ || (UpdateResourceNotification$ = {}));
export function updateResourceNotificationToJSON(updateResourceNotification) {
    return JSON.stringify(UpdateResourceNotification$outboundSchema.parse(updateResourceNotification));
}
export function updateResourceNotificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceNotification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceNotification' from JSON`);
}
/** @internal */
export const UpdateResourceEnvironmentOverrides$inboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
/** @internal */
export const UpdateResourceEnvironmentOverrides$outboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceEnvironmentOverrides$;
(function (UpdateResourceEnvironmentOverrides$) {
    /** @deprecated use `UpdateResourceEnvironmentOverrides$inboundSchema` instead. */
    UpdateResourceEnvironmentOverrides$.inboundSchema = UpdateResourceEnvironmentOverrides$inboundSchema;
    /** @deprecated use `UpdateResourceEnvironmentOverrides$outboundSchema` instead. */
    UpdateResourceEnvironmentOverrides$.outboundSchema = UpdateResourceEnvironmentOverrides$outboundSchema;
})(UpdateResourceEnvironmentOverrides$ || (UpdateResourceEnvironmentOverrides$ = {}));
export function updateResourceEnvironmentOverridesToJSON(updateResourceEnvironmentOverrides) {
    return JSON.stringify(UpdateResourceEnvironmentOverrides$outboundSchema.parse(updateResourceEnvironmentOverrides));
}
export function updateResourceEnvironmentOverridesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceEnvironmentOverrides$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceEnvironmentOverrides' from JSON`);
}
/** @internal */
export const UpdateResourceSecrets$inboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => UpdateResourceEnvironmentOverrides$inboundSchema).optional(),
});
/** @internal */
export const UpdateResourceSecrets$outboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => UpdateResourceEnvironmentOverrides$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceSecrets$;
(function (UpdateResourceSecrets$) {
    /** @deprecated use `UpdateResourceSecrets$inboundSchema` instead. */
    UpdateResourceSecrets$.inboundSchema = UpdateResourceSecrets$inboundSchema;
    /** @deprecated use `UpdateResourceSecrets$outboundSchema` instead. */
    UpdateResourceSecrets$.outboundSchema = UpdateResourceSecrets$outboundSchema;
})(UpdateResourceSecrets$ || (UpdateResourceSecrets$ = {}));
export function updateResourceSecretsToJSON(updateResourceSecrets) {
    return JSON.stringify(UpdateResourceSecrets$outboundSchema.parse(updateResourceSecrets));
}
export function updateResourceSecretsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecrets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecrets' from JSON`);
}
/** @internal */
export const UpdateResourceRequestBody$inboundSchema = z.object({
    ownership: UpdateResourceOwnership$inboundSchema.optional(),
    name: z.string().optional(),
    status: UpdateResourceStatus$inboundSchema.optional(),
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => UpdateResourceBillingPlan$inboundSchema).optional(),
    notification: z.lazy(() => UpdateResourceNotification$inboundSchema)
        .optional(),
    extras: z.record(z.any()).optional(),
    secrets: z.array(z.lazy(() => UpdateResourceSecrets$inboundSchema))
        .optional(),
});
/** @internal */
export const UpdateResourceRequestBody$outboundSchema = z.object({
    ownership: UpdateResourceOwnership$outboundSchema.optional(),
    name: z.string().optional(),
    status: UpdateResourceStatus$outboundSchema.optional(),
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => UpdateResourceBillingPlan$outboundSchema)
        .optional(),
    notification: z.lazy(() => UpdateResourceNotification$outboundSchema)
        .optional(),
    extras: z.record(z.any()).optional(),
    secrets: z.array(z.lazy(() => UpdateResourceSecrets$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceRequestBody$;
(function (UpdateResourceRequestBody$) {
    /** @deprecated use `UpdateResourceRequestBody$inboundSchema` instead. */
    UpdateResourceRequestBody$.inboundSchema = UpdateResourceRequestBody$inboundSchema;
    /** @deprecated use `UpdateResourceRequestBody$outboundSchema` instead. */
    UpdateResourceRequestBody$.outboundSchema = UpdateResourceRequestBody$outboundSchema;
})(UpdateResourceRequestBody$ || (UpdateResourceRequestBody$ = {}));
export function updateResourceRequestBodyToJSON(updateResourceRequestBody) {
    return JSON.stringify(UpdateResourceRequestBody$outboundSchema.parse(updateResourceRequestBody));
}
export function updateResourceRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceRequestBody' from JSON`);
}
/** @internal */
export const UpdateResourceRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
    RequestBody: z.lazy(() => UpdateResourceRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateResourceRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
    requestBody: z.lazy(() => UpdateResourceRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceRequest$;
(function (UpdateResourceRequest$) {
    /** @deprecated use `UpdateResourceRequest$inboundSchema` instead. */
    UpdateResourceRequest$.inboundSchema = UpdateResourceRequest$inboundSchema;
    /** @deprecated use `UpdateResourceRequest$outboundSchema` instead. */
    UpdateResourceRequest$.outboundSchema = UpdateResourceRequest$outboundSchema;
})(UpdateResourceRequest$ || (UpdateResourceRequest$ = {}));
export function updateResourceRequestToJSON(updateResourceRequest) {
    return JSON.stringify(UpdateResourceRequest$outboundSchema.parse(updateResourceRequest));
}
export function updateResourceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceRequest' from JSON`);
}
/** @internal */
export const UpdateResourceResponseBody$inboundSchema = z.object({
    name: z.string(),
});
/** @internal */
export const UpdateResourceResponseBody$outboundSchema = z.object({
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateResourceResponseBody$;
(function (UpdateResourceResponseBody$) {
    /** @deprecated use `UpdateResourceResponseBody$inboundSchema` instead. */
    UpdateResourceResponseBody$.inboundSchema = UpdateResourceResponseBody$inboundSchema;
    /** @deprecated use `UpdateResourceResponseBody$outboundSchema` instead. */
    UpdateResourceResponseBody$.outboundSchema = UpdateResourceResponseBody$outboundSchema;
})(UpdateResourceResponseBody$ || (UpdateResourceResponseBody$ = {}));
export function updateResourceResponseBodyToJSON(updateResourceResponseBody) {
    return JSON.stringify(UpdateResourceResponseBody$outboundSchema.parse(updateResourceResponseBody));
}
export function updateResourceResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceResponseBody' from JSON`);
}
