// @vercel/sdk/esm/models/updatesharedconnectlinksop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateSharedConnectLinksEnvId2 = {
    Preview: "preview",
    Production: "production",
};
/** @internal */
export const UpdateSharedConnectLinksRequestBody$inboundSchema = z.object({
    regions: z.array(z.string()),
});
/** @internal */
export const UpdateSharedConnectLinksRequestBody$outboundSchema = z.object({
    regions: z.array(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateSharedConnectLinksRequestBody$;
(function (UpdateSharedConnectLinksRequestBody$) {
    /** @deprecated use `UpdateSharedConnectLinksRequestBody$inboundSchema` instead. */
    UpdateSharedConnectLinksRequestBody$.inboundSchema = UpdateSharedConnectLinksRequestBody$inboundSchema;
    /** @deprecated use `UpdateSharedConnectLinksRequestBody$outboundSchema` instead. */
    UpdateSharedConnectLinksRequestBody$.outboundSchema = UpdateSharedConnectLinksRequestBody$outboundSchema;
})(UpdateSharedConnectLinksRequestBody$ || (UpdateSharedConnectLinksRequestBody$ = {}));
export function updateSharedConnectLinksRequestBodyToJSON(updateSharedConnectLinksRequestBody) {
    return JSON.stringify(UpdateSharedConnectLinksRequestBody$outboundSchema.parse(updateSharedConnectLinksRequestBody));
}
export function updateSharedConnectLinksRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedConnectLinksRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedConnectLinksRequestBody' from JSON`);
}
/** @internal */
export const UpdateSharedConnectLinksRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateSharedConnectLinksRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateSharedConnectLinksRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateSharedConnectLinksRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateSharedConnectLinksRequest$;
(function (UpdateSharedConnectLinksRequest$) {
    /** @deprecated use `UpdateSharedConnectLinksRequest$inboundSchema` instead. */
    UpdateSharedConnectLinksRequest$.inboundSchema = UpdateSharedConnectLinksRequest$inboundSchema;
    /** @deprecated use `UpdateSharedConnectLinksRequest$outboundSchema` instead. */
    UpdateSharedConnectLinksRequest$.outboundSchema = UpdateSharedConnectLinksRequest$outboundSchema;
})(UpdateSharedConnectLinksRequest$ || (UpdateSharedConnectLinksRequest$ = {}));
export function updateSharedConnectLinksRequestToJSON(updateSharedConnectLinksRequest) {
    return JSON.stringify(UpdateSharedConnectLinksRequest$outboundSchema.parse(updateSharedConnectLinksRequest));
}
export function updateSharedConnectLinksRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedConnectLinksRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedConnectLinksRequest' from JSON`);
}
/** @internal */
export const UpdateSharedConnectLinksEnvId2$inboundSchema = z.nativeEnum(UpdateSharedConnectLinksEnvId2);
/** @internal */
export const UpdateSharedConnectLinksEnvId2$outboundSchema = UpdateSharedConnectLinksEnvId2$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateSharedConnectLinksEnvId2$;
(function (UpdateSharedConnectLinksEnvId2$) {
    /** @deprecated use `UpdateSharedConnectLinksEnvId2$inboundSchema` instead. */
    UpdateSharedConnectLinksEnvId2$.inboundSchema = UpdateSharedConnectLinksEnvId2$inboundSchema;
    /** @deprecated use `UpdateSharedConnectLinksEnvId2$outboundSchema` instead. */
    UpdateSharedConnectLinksEnvId2$.outboundSchema = UpdateSharedConnectLinksEnvId2$outboundSchema;
})(UpdateSharedConnectLinksEnvId2$ || (UpdateSharedConnectLinksEnvId2$ = {}));
/** @internal */
export const UpdateSharedConnectLinksEnvId$inboundSchema = z.union([z.string(), UpdateSharedConnectLinksEnvId2$inboundSchema]);
/** @internal */
export const UpdateSharedConnectLinksEnvId$outboundSchema = z.union([z.string(), UpdateSharedConnectLinksEnvId2$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateSharedConnectLinksEnvId$;
(function (UpdateSharedConnectLinksEnvId$) {
    /** @deprecated use `UpdateSharedConnectLinksEnvId$inboundSchema` instead. */
    UpdateSharedConnectLinksEnvId$.inboundSchema = UpdateSharedConnectLinksEnvId$inboundSchema;
    /** @deprecated use `UpdateSharedConnectLinksEnvId$outboundSchema` instead. */
    UpdateSharedConnectLinksEnvId$.outboundSchema = UpdateSharedConnectLinksEnvId$outboundSchema;
})(UpdateSharedConnectLinksEnvId$ || (UpdateSharedConnectLinksEnvId$ = {}));
export function updateSharedConnectLinksEnvIdToJSON(updateSharedConnectLinksEnvId) {
    return JSON.stringify(UpdateSharedConnectLinksEnvId$outboundSchema.parse(updateSharedConnectLinksEnvId));
}
export function updateSharedConnectLinksEnvIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedConnectLinksEnvId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedConnectLinksEnvId' from JSON`);
}
/** @internal */
export const UpdateSharedConnectLinksAws$inboundSchema = z.object({
    subnetIds: z.array(z.string()),
    securityGroupId: z.string(),
});
/** @internal */
export const UpdateSharedConnectLinksAws$outboundSchema = z.object({
    subnetIds: z.array(z.string()),
    securityGroupId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateSharedConnectLinksAws$;
(function (UpdateSharedConnectLinksAws$) {
    /** @deprecated use `UpdateSharedConnectLinksAws$inboundSchema` instead. */
    UpdateSharedConnectLinksAws$.inboundSchema = UpdateSharedConnectLinksAws$inboundSchema;
    /** @deprecated use `UpdateSharedConnectLinksAws$outboundSchema` instead. */
    UpdateSharedConnectLinksAws$.outboundSchema = UpdateSharedConnectLinksAws$outboundSchema;
})(UpdateSharedConnectLinksAws$ || (UpdateSharedConnectLinksAws$ = {}));
export function updateSharedConnectLinksAwsToJSON(updateSharedConnectLinksAws) {
    return JSON.stringify(UpdateSharedConnectLinksAws$outboundSchema.parse(updateSharedConnectLinksAws));
}
export function updateSharedConnectLinksAwsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedConnectLinksAws$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedConnectLinksAws' from JSON`);
}
/** @internal */
export const UpdateSharedConnectLinksResponseBody$inboundSchema = z.object({
    envId: z.union([z.string(), UpdateSharedConnectLinksEnvId2$inboundSchema]),
    connectConfigurationId: z.string(),
    dc: z.string().optional(),
    passive: z.boolean(),
    buildsEnabled: z.boolean(),
    aws: z.lazy(() => UpdateSharedConnectLinksAws$inboundSchema).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const UpdateSharedConnectLinksResponseBody$outboundSchema = z.object({
    envId: z.union([z.string(), UpdateSharedConnectLinksEnvId2$outboundSchema]),
    connectConfigurationId: z.string(),
    dc: z.string().optional(),
    passive: z.boolean(),
    buildsEnabled: z.boolean(),
    aws: z.lazy(() => UpdateSharedConnectLinksAws$outboundSchema).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateSharedConnectLinksResponseBody$;
(function (UpdateSharedConnectLinksResponseBody$) {
    /** @deprecated use `UpdateSharedConnectLinksResponseBody$inboundSchema` instead. */
    UpdateSharedConnectLinksResponseBody$.inboundSchema = UpdateSharedConnectLinksResponseBody$inboundSchema;
    /** @deprecated use `UpdateSharedConnectLinksResponseBody$outboundSchema` instead. */
    UpdateSharedConnectLinksResponseBody$.outboundSchema = UpdateSharedConnectLinksResponseBody$outboundSchema;
})(UpdateSharedConnectLinksResponseBody$ || (UpdateSharedConnectLinksResponseBody$ = {}));
export function updateSharedConnectLinksResponseBodyToJSON(updateSharedConnectLinksResponseBody) {
    return JSON.stringify(UpdateSharedConnectLinksResponseBody$outboundSchema.parse(updateSharedConnectLinksResponseBody));
}
export function updateSharedConnectLinksResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedConnectLinksResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedConnectLinksResponseBody' from JSON`);
}
