// @vercel/sdk/esm/models/updatedomainautorenewop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { DomainAlreadyRenewing$inboundSchema, DomainAlreadyRenewing$outboundSchema, } from "./domainalreadyrenewing.js";
import { DomainNotRegistered$inboundSchema, DomainNotRegistered$outboundSchema, } from "./domainnotregistered.js";
import { DomainNotRenewable$inboundSchema, DomainNotRenewable$outboundSchema, } from "./domainnotrenewable.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
/** @internal */
export const UpdateDomainAutoRenewRequestBody$inboundSchema = z.object({
    autoRenew: z.boolean(),
});
/** @internal */
export const UpdateDomainAutoRenewRequestBody$outboundSchema = z.object({
    autoRenew: z.boolean(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainAutoRenewRequestBody$;
(function (UpdateDomainAutoRenewRequestBody$) {
    /** @deprecated use `UpdateDomainAutoRenewRequestBody$inboundSchema` instead. */
    UpdateDomainAutoRenewRequestBody$.inboundSchema = UpdateDomainAutoRenewRequestBody$inboundSchema;
    /** @deprecated use `UpdateDomainAutoRenewRequestBody$outboundSchema` instead. */
    UpdateDomainAutoRenewRequestBody$.outboundSchema = UpdateDomainAutoRenewRequestBody$outboundSchema;
})(UpdateDomainAutoRenewRequestBody$ || (UpdateDomainAutoRenewRequestBody$ = {}));
export function updateDomainAutoRenewRequestBodyToJSON(updateDomainAutoRenewRequestBody) {
    return JSON.stringify(UpdateDomainAutoRenewRequestBody$outboundSchema.parse(updateDomainAutoRenewRequestBody));
}
export function updateDomainAutoRenewRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainAutoRenewRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainAutoRenewRequestBody' from JSON`);
}
/** @internal */
export const UpdateDomainAutoRenewRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    RequestBody: z.lazy(() => UpdateDomainAutoRenewRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateDomainAutoRenewRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    requestBody: z.lazy(() => UpdateDomainAutoRenewRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainAutoRenewRequest$;
(function (UpdateDomainAutoRenewRequest$) {
    /** @deprecated use `UpdateDomainAutoRenewRequest$inboundSchema` instead. */
    UpdateDomainAutoRenewRequest$.inboundSchema = UpdateDomainAutoRenewRequest$inboundSchema;
    /** @deprecated use `UpdateDomainAutoRenewRequest$outboundSchema` instead. */
    UpdateDomainAutoRenewRequest$.outboundSchema = UpdateDomainAutoRenewRequest$outboundSchema;
})(UpdateDomainAutoRenewRequest$ || (UpdateDomainAutoRenewRequest$ = {}));
export function updateDomainAutoRenewRequestToJSON(updateDomainAutoRenewRequest) {
    return JSON.stringify(UpdateDomainAutoRenewRequest$outboundSchema.parse(updateDomainAutoRenewRequest));
}
export function updateDomainAutoRenewRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainAutoRenewRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainAutoRenewRequest' from JSON`);
}
/** @internal */
export const UpdateDomainAutoRenewDomainsRegistrarResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const UpdateDomainAutoRenewDomainsRegistrarResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainAutoRenewDomainsRegistrarResponseBody$;
(function (UpdateDomainAutoRenewDomainsRegistrarResponseBody$) {
    /** @deprecated use `UpdateDomainAutoRenewDomainsRegistrarResponseBody$inboundSchema` instead. */
    UpdateDomainAutoRenewDomainsRegistrarResponseBody$.inboundSchema = UpdateDomainAutoRenewDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `UpdateDomainAutoRenewDomainsRegistrarResponseBody$outboundSchema` instead. */
    UpdateDomainAutoRenewDomainsRegistrarResponseBody$.outboundSchema = UpdateDomainAutoRenewDomainsRegistrarResponseBody$outboundSchema;
})(UpdateDomainAutoRenewDomainsRegistrarResponseBody$ || (UpdateDomainAutoRenewDomainsRegistrarResponseBody$ = {}));
export function updateDomainAutoRenewDomainsRegistrarResponseBodyToJSON(updateDomainAutoRenewDomainsRegistrarResponseBody) {
    return JSON.stringify(UpdateDomainAutoRenewDomainsRegistrarResponseBody$outboundSchema.parse(updateDomainAutoRenewDomainsRegistrarResponseBody));
}
export function updateDomainAutoRenewDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainAutoRenewDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainAutoRenewDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const UpdateDomainAutoRenewResponseBody$inboundSchema = z.union([
    DomainAlreadyRenewing$inboundSchema,
    DomainNotRenewable$inboundSchema,
    DomainNotRegistered$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const UpdateDomainAutoRenewResponseBody$outboundSchema = z.union([
    DomainAlreadyRenewing$outboundSchema,
    DomainNotRenewable$outboundSchema,
    DomainNotRegistered$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainAutoRenewResponseBody$;
(function (UpdateDomainAutoRenewResponseBody$) {
    /** @deprecated use `UpdateDomainAutoRenewResponseBody$inboundSchema` instead. */
    UpdateDomainAutoRenewResponseBody$.inboundSchema = UpdateDomainAutoRenewResponseBody$inboundSchema;
    /** @deprecated use `UpdateDomainAutoRenewResponseBody$outboundSchema` instead. */
    UpdateDomainAutoRenewResponseBody$.outboundSchema = UpdateDomainAutoRenewResponseBody$outboundSchema;
})(UpdateDomainAutoRenewResponseBody$ || (UpdateDomainAutoRenewResponseBody$ = {}));
export function updateDomainAutoRenewResponseBodyToJSON(updateDomainAutoRenewResponseBody) {
    return JSON.stringify(UpdateDomainAutoRenewResponseBody$outboundSchema.parse(updateDomainAutoRenewResponseBody));
}
export function updateDomainAutoRenewResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainAutoRenewResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainAutoRenewResponseBody' from JSON`);
}
