// @vercel/sdk/esm/models/updateinstallationop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
export const UpdateInstallationType = {
    Prepayment: "prepayment",
    Subscription: "subscription",
};
export const Level = {
    Info: "info",
    Warn: "warn",
    Error: "error",
};
/** @internal */
export const UpdateInstallationType$inboundSchema = z.nativeEnum(UpdateInstallationType);
/** @internal */
export const UpdateInstallationType$outboundSchema = UpdateInstallationType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateInstallationType$;
(function (UpdateInstallationType$) {
    /** @deprecated use `UpdateInstallationType$inboundSchema` instead. */
    UpdateInstallationType$.inboundSchema = UpdateInstallationType$inboundSchema;
    /** @deprecated use `UpdateInstallationType$outboundSchema` instead. */
    UpdateInstallationType$.outboundSchema = UpdateInstallationType$outboundSchema;
})(UpdateInstallationType$ || (UpdateInstallationType$ = {}));
/** @internal */
export const Details$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const Details$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Details$;
(function (Details$) {
    /** @deprecated use `Details$inboundSchema` instead. */
    Details$.inboundSchema = Details$inboundSchema;
    /** @deprecated use `Details$outboundSchema` instead. */
    Details$.outboundSchema = Details$outboundSchema;
})(Details$ || (Details$ = {}));
export function detailsToJSON(details) {
    return JSON.stringify(Details$outboundSchema.parse(details));
}
export function detailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Details$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Details' from JSON`);
}
/** @internal */
export const HighlightedDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const HighlightedDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var HighlightedDetails$;
(function (HighlightedDetails$) {
    /** @deprecated use `HighlightedDetails$inboundSchema` instead. */
    HighlightedDetails$.inboundSchema = HighlightedDetails$inboundSchema;
    /** @deprecated use `HighlightedDetails$outboundSchema` instead. */
    HighlightedDetails$.outboundSchema = HighlightedDetails$outboundSchema;
})(HighlightedDetails$ || (HighlightedDetails$ = {}));
export function highlightedDetailsToJSON(highlightedDetails) {
    return JSON.stringify(HighlightedDetails$outboundSchema.parse(highlightedDetails));
}
export function highlightedDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => HighlightedDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'HighlightedDetails' from JSON`);
}
/** @internal */
export const BillingPlan$inboundSchema = collectExtraKeys$(z.object({
    id: z.string(),
    type: UpdateInstallationType$inboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => Details$inboundSchema)).optional(),
    highlightedDetails: z.array(z.lazy(() => HighlightedDetails$inboundSchema))
        .optional(),
    effectiveDate: z.string().optional(),
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const BillingPlan$outboundSchema = z.object({
    id: z.string(),
    type: UpdateInstallationType$outboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => Details$outboundSchema)).optional(),
    highlightedDetails: z.array(z.lazy(() => HighlightedDetails$outboundSchema))
        .optional(),
    effectiveDate: z.string().optional(),
    additionalProperties: z.record(z.any()),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BillingPlan$;
(function (BillingPlan$) {
    /** @deprecated use `BillingPlan$inboundSchema` instead. */
    BillingPlan$.inboundSchema = BillingPlan$inboundSchema;
    /** @deprecated use `BillingPlan$outboundSchema` instead. */
    BillingPlan$.outboundSchema = BillingPlan$outboundSchema;
})(BillingPlan$ || (BillingPlan$ = {}));
export function billingPlanToJSON(billingPlan) {
    return JSON.stringify(BillingPlan$outboundSchema.parse(billingPlan));
}
export function billingPlanFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BillingPlan$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BillingPlan' from JSON`);
}
/** @internal */
export const Level$inboundSchema = z.nativeEnum(Level);
/** @internal */
export const Level$outboundSchema = Level$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Level$;
(function (Level$) {
    /** @deprecated use `Level$inboundSchema` instead. */
    Level$.inboundSchema = Level$inboundSchema;
    /** @deprecated use `Level$outboundSchema` instead. */
    Level$.outboundSchema = Level$outboundSchema;
})(Level$ || (Level$ = {}));
/** @internal */
export const Notification$inboundSchema = z.object({
    level: Level$inboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/** @internal */
export const Notification$outboundSchema = z.object({
    level: Level$outboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Notification$;
(function (Notification$) {
    /** @deprecated use `Notification$inboundSchema` instead. */
    Notification$.inboundSchema = Notification$inboundSchema;
    /** @deprecated use `Notification$outboundSchema` instead. */
    Notification$.outboundSchema = Notification$outboundSchema;
})(Notification$ || (Notification$ = {}));
export function notificationToJSON(notification) {
    return JSON.stringify(Notification$outboundSchema.parse(notification));
}
export function notificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Notification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Notification' from JSON`);
}
/** @internal */
export const UpdateInstallationRequestBody$inboundSchema = z.object({
    billingPlan: z.lazy(() => BillingPlan$inboundSchema).optional(),
    notification: z.lazy(() => Notification$inboundSchema).optional(),
});
/** @internal */
export const UpdateInstallationRequestBody$outboundSchema = z.object({
    billingPlan: z.lazy(() => BillingPlan$outboundSchema).optional(),
    notification: z.lazy(() => Notification$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateInstallationRequestBody$;
(function (UpdateInstallationRequestBody$) {
    /** @deprecated use `UpdateInstallationRequestBody$inboundSchema` instead. */
    UpdateInstallationRequestBody$.inboundSchema = UpdateInstallationRequestBody$inboundSchema;
    /** @deprecated use `UpdateInstallationRequestBody$outboundSchema` instead. */
    UpdateInstallationRequestBody$.outboundSchema = UpdateInstallationRequestBody$outboundSchema;
})(UpdateInstallationRequestBody$ || (UpdateInstallationRequestBody$ = {}));
export function updateInstallationRequestBodyToJSON(updateInstallationRequestBody) {
    return JSON.stringify(UpdateInstallationRequestBody$outboundSchema.parse(updateInstallationRequestBody));
}
export function updateInstallationRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateInstallationRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateInstallationRequestBody' from JSON`);
}
/** @internal */
export const UpdateInstallationRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    RequestBody: z.lazy(() => UpdateInstallationRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateInstallationRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    requestBody: z.lazy(() => UpdateInstallationRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateInstallationRequest$;
(function (UpdateInstallationRequest$) {
    /** @deprecated use `UpdateInstallationRequest$inboundSchema` instead. */
    UpdateInstallationRequest$.inboundSchema = UpdateInstallationRequest$inboundSchema;
    /** @deprecated use `UpdateInstallationRequest$outboundSchema` instead. */
    UpdateInstallationRequest$.outboundSchema = UpdateInstallationRequest$outboundSchema;
})(UpdateInstallationRequest$ || (UpdateInstallationRequest$ = {}));
export function updateInstallationRequestToJSON(updateInstallationRequest) {
    return JSON.stringify(UpdateInstallationRequest$outboundSchema.parse(updateInstallationRequest));
}
export function updateInstallationRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateInstallationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateInstallationRequest' from JSON`);
}
