// @vercel/sdk/esm/models/updatedomainnameserversop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { DomainNotRegistered$inboundSchema, DomainNotRegistered$outboundSchema, } from "./domainnotregistered.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
/** @internal */
export const UpdateDomainNameserversRequestBody$inboundSchema = z.object({
    nameservers: z.array(z.string()),
});
/** @internal */
export const UpdateDomainNameserversRequestBody$outboundSchema = z.object({
    nameservers: z.array(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainNameserversRequestBody$;
(function (UpdateDomainNameserversRequestBody$) {
    /** @deprecated use `UpdateDomainNameserversRequestBody$inboundSchema` instead. */
    UpdateDomainNameserversRequestBody$.inboundSchema = UpdateDomainNameserversRequestBody$inboundSchema;
    /** @deprecated use `UpdateDomainNameserversRequestBody$outboundSchema` instead. */
    UpdateDomainNameserversRequestBody$.outboundSchema = UpdateDomainNameserversRequestBody$outboundSchema;
})(UpdateDomainNameserversRequestBody$ || (UpdateDomainNameserversRequestBody$ = {}));
export function updateDomainNameserversRequestBodyToJSON(updateDomainNameserversRequestBody) {
    return JSON.stringify(UpdateDomainNameserversRequestBody$outboundSchema.parse(updateDomainNameserversRequestBody));
}
export function updateDomainNameserversRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainNameserversRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainNameserversRequestBody' from JSON`);
}
/** @internal */
export const UpdateDomainNameserversRequest$inboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    RequestBody: z.lazy(() => UpdateDomainNameserversRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateDomainNameserversRequest$outboundSchema = z.object({
    domain: z.string(),
    teamId: z.string().optional(),
    requestBody: z.lazy(() => UpdateDomainNameserversRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainNameserversRequest$;
(function (UpdateDomainNameserversRequest$) {
    /** @deprecated use `UpdateDomainNameserversRequest$inboundSchema` instead. */
    UpdateDomainNameserversRequest$.inboundSchema = UpdateDomainNameserversRequest$inboundSchema;
    /** @deprecated use `UpdateDomainNameserversRequest$outboundSchema` instead. */
    UpdateDomainNameserversRequest$.outboundSchema = UpdateDomainNameserversRequest$outboundSchema;
})(UpdateDomainNameserversRequest$ || (UpdateDomainNameserversRequest$ = {}));
export function updateDomainNameserversRequestToJSON(updateDomainNameserversRequest) {
    return JSON.stringify(UpdateDomainNameserversRequest$outboundSchema.parse(updateDomainNameserversRequest));
}
export function updateDomainNameserversRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainNameserversRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainNameserversRequest' from JSON`);
}
/** @internal */
export const UpdateDomainNameserversDomainsRegistrarResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const UpdateDomainNameserversDomainsRegistrarResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainNameserversDomainsRegistrarResponseBody$;
(function (UpdateDomainNameserversDomainsRegistrarResponseBody$) {
    /** @deprecated use `UpdateDomainNameserversDomainsRegistrarResponseBody$inboundSchema` instead. */
    UpdateDomainNameserversDomainsRegistrarResponseBody$.inboundSchema = UpdateDomainNameserversDomainsRegistrarResponseBody$inboundSchema;
    /** @deprecated use `UpdateDomainNameserversDomainsRegistrarResponseBody$outboundSchema` instead. */
    UpdateDomainNameserversDomainsRegistrarResponseBody$.outboundSchema = UpdateDomainNameserversDomainsRegistrarResponseBody$outboundSchema;
})(UpdateDomainNameserversDomainsRegistrarResponseBody$ || (UpdateDomainNameserversDomainsRegistrarResponseBody$ = {}));
export function updateDomainNameserversDomainsRegistrarResponseBodyToJSON(updateDomainNameserversDomainsRegistrarResponseBody) {
    return JSON.stringify(UpdateDomainNameserversDomainsRegistrarResponseBody$outboundSchema.parse(updateDomainNameserversDomainsRegistrarResponseBody));
}
export function updateDomainNameserversDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainNameserversDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainNameserversDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const UpdateDomainNameserversResponseBody$inboundSchema = z.union([
    DomainNotRegistered$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const UpdateDomainNameserversResponseBody$outboundSchema = z.union([
    DomainNotRegistered$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateDomainNameserversResponseBody$;
(function (UpdateDomainNameserversResponseBody$) {
    /** @deprecated use `UpdateDomainNameserversResponseBody$inboundSchema` instead. */
    UpdateDomainNameserversResponseBody$.inboundSchema = UpdateDomainNameserversResponseBody$inboundSchema;
    /** @deprecated use `UpdateDomainNameserversResponseBody$outboundSchema` instead. */
    UpdateDomainNameserversResponseBody$.outboundSchema = UpdateDomainNameserversResponseBody$outboundSchema;
})(UpdateDomainNameserversResponseBody$ || (UpdateDomainNameserversResponseBody$ = {}));
export function updateDomainNameserversResponseBodyToJSON(updateDomainNameserversResponseBody) {
    return JSON.stringify(UpdateDomainNameserversResponseBody$outboundSchema.parse(updateDomainNameserversResponseBody));
}
export function updateDomainNameserversResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDomainNameserversResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDomainNameserversResponseBody' from JSON`);
}
